#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff --git a/src/plugins/pgpinline/claws.def b/src/plugins/pgpinline/claws.def
index 30f963e24..db3a934be 100644
--- a/src/plugins/pgpinline/claws.def
+++ b/src/plugins/pgpinline/claws.def
@@ -12,6 +12,7 @@ conv_get_locale_charset_str_no_utf8
 debug_print_real
 debug_srcname
 get_mime_tmp_dir
+file_read_stream_to_str_no_recode
 my_tmpfile
 privacy_register_system
 privacy_reset_error
diff --git a/src/plugins/rssyl/claws.def b/src/plugins/rssyl/claws.def
index 85de45e36..a36841d42 100644
--- a/src/plugins/rssyl/claws.def
+++ b/src/plugins/rssyl/claws.def
@@ -118,6 +118,7 @@ prefs_write_param
 procheader_date_parse
 procheader_parse_file
 procmsg_get_message_file
+procmsg_msginfo_free
 procmsg_msginfo_unset_flags
 remove_dir_recursive
 slist_free_strings_full
diff --git a/src/plugins/smime/claws.def b/src/plugins/smime/claws.def
index a3eeed5ef..351805937 100644
--- a/src/plugins/smime/claws.def
+++ b/src/plugins/smime/claws.def
@@ -12,6 +12,7 @@ claws_unlink
 debug_print_real
 debug_srcname
 file_read_stream_to_str
+file_read_stream_to_str_no_recode
 file_read_to_str
 generate_mime_boundary
 get_mime_tmp_dir
diff --git a/src/plugins/vcalendar/claws.def b/src/plugins/vcalendar/claws.def
index 9022e2404..419b19d1e 100644
--- a/src/plugins/vcalendar/claws.def
+++ b/src/plugins/vcalendar/claws.def
@@ -141,6 +141,7 @@ procheader_parse_file
 procmime_get_first_encrypted_text_content
 procmime_get_first_text_content
 procmime_get_part
+procmime_get_part_as_string
 procmime_get_tmp_file_name
 procmime_mimeinfo_get_parameter
 procmsg_msginfo_free
