#! /bin/sh
patch -p1 -f $* < $0
exit $?

--- claws-mail-3.17.4-orig/src/plugins/litehtml_viewer/Makefile.in	2019-07-26 13:01:27.000000000 +0200
+++ claws-mail-3.17.4/src/plugins/litehtml_viewer/Makefile.in	2019-08-25 18:55:00.274314999 +0200
@@ -544,6 +544,7 @@
 psdir = @psdir@
 pyexecdir = @pyexecdir@
 pythondir = @pythondir@
+runstatedir = @runstatedir@
 sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
@@ -565,18 +566,29 @@
         -I$(top_srcdir)/src/common \
         -I$(top_builddir)/src \
         -I$(top_srcdir)/src/gtk \
-	-I$(top_srcdir)/src/plugins/litehtml_viewer/litehtml
+        -I$(top_srcdir)/src/plugins/litehtml_viewer/litehtml
 
-plugin_res = 
-plugin_res_ldflag = 
-export_symbols = 
-plugin_deps = 
-plugin_ldadd = 
-no_undefined = 
-cygwin_export_lib = 
+@OS_WIN32_TRUE@LTRCCOMPILE = $(LIBTOOL) --mode=compile --tag=RC $(RC) \
+@OS_WIN32_TRUE@		`echo $(DEFS) $(DEFAULT_INCLUDES) $(IFLAGS) | \
+@OS_WIN32_TRUE@		sed -e 's/-I/--include-dir /g;s/-D/--define /g'`
+
+@OS_WIN32_FALSE@plugin_res = 
+@OS_WIN32_TRUE@plugin_res = version.lo
+@OS_WIN32_FALSE@plugin_res_ldflag = 
+@OS_WIN32_TRUE@plugin_res_ldflag = -Wl,.libs/version.o
+@OS_WIN32_FALSE@export_symbols = 
+@OS_WIN32_TRUE@export_symbols = -export-symbols $(srcdir)/plugin.def
+@OS_WIN32_FALSE@plugin_deps = 
+@OS_WIN32_TRUE@plugin_deps = libclaws.a $(plugin_res) plugin.def
+@OS_WIN32_FALSE@plugin_ldadd = 
+@OS_WIN32_TRUE@plugin_ldadd = -L. -lclaws
+@PLATFORM_WIN32_FALSE@no_undefined = 
+@PLATFORM_WIN32_TRUE@no_undefined = -no-undefined
+@CYGWIN_FALSE@cygwin_export_lib = 
+@CYGWIN_TRUE@cygwin_export_lib = -L$(top_builddir)/src -lclaws-mail
 plugindir = $(pkglibdir)/plugins
 @BUILD_LITEHTML_VIEWER_PLUGIN_TRUE@plugin_LTLIBRARIES = litehtml_viewer.la
-litehtml_viewer_la_DEPENDENCIES = $(plugin_deps)
+litehtml_viewer_la_DEPENDENCIES = $(plugin_deps) litehtml/liblitehtml.la
 litehtml_viewer_la_LIBADD = $(plugin_ldadd) $(cygwin_export_lib) \
 	$(top_builddir)/src/plugins/litehtml_viewer/litehtml/liblitehtml.la \
         $(GTK_LIBS)
@@ -1116,6 +1128,12 @@
 .PRECIOUS: Makefile
 
 
+@OS_WIN32_TRUE@%.lo: %.rc
+@OS_WIN32_TRUE@	$(LTRCCOMPILE) -i $< -o $@
+
+@OS_WIN32_TRUE@libclaws.a: claws.def
+@OS_WIN32_TRUE@	$(DLLTOOL) --output-lib $@ --def $<
+
 .PHONY: test
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
