#! /bin/sh
patch -p1 -f $* < $0
exit $?

--- claws-mail-3.17.4-orig/src/plugins/litehtml_viewer/litehtml/os_types.h	2019-02-13 09:32:44.000000000 +0100
+++ claws-mail-3.17.4/src/plugins/litehtml_viewer/litehtml/os_types.h	2019-08-25 18:57:30.587708334 +0200
@@ -49,11 +49,12 @@
 
 #endif
 
-	#ifdef _WIN64
-		typedef unsigned __int64 uint_ptr;
-	#else
-		typedef unsigned int	uint_ptr;
-	#endif
+//	#ifdef _WIN64
+//		typedef unsigned __int64 uint_ptr;
+//	#else
+//		typedef unsigned int	uint_ptr;
+//	#endif
+	typedef void*				uint_ptr;
 
 #else
 	#define LITEHTML_UTF8
