#!/bin/sh
patch -p1 -f $* < $0
exit $?

diff -urN a/bzlib.h b/bzlib.h
--- a/bzlib.h	2010-09-10 16:08:42.000000000 -0700
+++ b/bzlib.h	2010-09-23 00:04:43.000000000 -0700
@@ -75,7 +75,9 @@
 #include <stdio.h>
 #endif
 
-#ifdef _WIN32
+/* We do not need a DLL and thus we don't want to include the
+   Windows header file.   (g10 Code, 2007-08-20) */
+#if defined( _WIN32 ) && 0
 #   include <windows.h>
 #   ifdef small
       /* windows.h define small to char */
