#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN webkitgtk-2.3.5-orig/Source/WTF/wtf/Platform.h webkitgtk-2.3.5/Source/WTF/wtf/Platform.h
--- webkitgtk-2.3.5-orig/Source/WTF/wtf/Platform.h	2016-12-01 23:55:54.194262026 +0100
+++ webkitgtk-2.3.5/Source/WTF/wtf/Platform.h	2016-12-01 23:56:21.917684891 +0100
@@ -594,8 +594,6 @@
 #define HAVE_ALIGNED_MALLOC 1
 #define HAVE_ISDEBUGGERPRESENT 1
 
-#include <WTF/WTFHeaderDetection.h>
-
 #endif
 
 #if OS(WINDOWS)
