#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN webkitgtk-2.3.5a-orig/Source/WTF/wtf/Atomics.h webkitgtk-2.3.5a/Source/WTF/wtf/Atomics.h
--- webkitgtk-2.3.5a-orig/Source/WTF/wtf/Atomics.h	2014-01-28 14:43:43.000000000 +0100
+++ webkitgtk-2.3.5a/Source/WTF/wtf/Atomics.h	2017-02-05 17:25:24.107771575 +0100
@@ -270,20 +270,6 @@
         : "memory"
         );
     return result;
-#elif OS(WINDOWS) && CPU(X86)
-    // FIXME: We need a 64-bit ASM implementation, but this cannot be inline due to
-    // Microsoft's decision to exclude it from the compiler.
-    bool result = false;
-
-    __asm {
-        mov al, expected
-        mov edx, location
-        mov cl, newValue
-        lock cmpxchg byte ptr[edx], cl
-        setz result
-    }
-
-    return result;
 #else
     uintptr_t locationValue = bitwise_cast<uintptr_t>(location);
     uintptr_t alignedLocationValue = locationValue & ~(sizeof(unsigned) - 1);
