#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN webkitgtk-2.3.5a-orig/configure webkitgtk-2.3.5a/configure
--- webkitgtk-2.3.5a-orig/configure	2014-02-05 12:07:47.000000000 +0100
+++ webkitgtk-2.3.5a/configure	2017-02-05 21:43:11.006430012 +0100
@@ -19047,22 +19047,6 @@
 $as_echo "$enable_gles2" >&6; }
 
 found_opengl=no
-if test "$enable_gles2" = "yes"; then
-    found_opengl=yes
-else
-    for ac_header in GL/gl.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "GL/gl.h" "ac_cv_header_GL_gl_h" "$ac_includes_default"
-if test "x$ac_cv_header_GL_gl_h" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_GL_GL_H 1
-_ACEOF
- found_opengl="yes"
-fi
-
-done
-
-fi
 
 if test "$enable_x11_target" = "yes" && test "$found_opengl" = "yes"; then
 
