#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN cyrus-sasl-2.1.26-orig/include/Makefile.am cyrus-sasl-2.1.26/include/Makefile.am
--- cyrus-sasl-2.1.26-orig/include/Makefile.am	2016-05-29 15:03:09.302579361 +0200
+++ cyrus-sasl-2.1.26/include/Makefile.am	2016-05-29 15:26:59.616935929 +0200
@@ -47,9 +47,8 @@
 saslincludedir = $(includedir)/sasl
 saslinclude_HEADERS = hmac-md5.h md5.h md5global.h sasl.h saslplug.h saslutil.h prop.h
 
-noinst_PROGRAMS = makemd5
-
-makemd5_SOURCES = makemd5.c
+makemd5:
+	gcc -o makemd5 ${srcdir}/makemd5.c
 
 md5global.h: makemd5
 	-rm -f md5global.h
Binary files cyrus-sasl-2.1.26-orig/include/makemd5 and cyrus-sasl-2.1.26/include/makemd5 differ
