#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN glib-2.56.2-orig/Makefile.in glib-2.56.2/Makefile.in
--- glib-2.56.2-orig/Makefile.in	2018-08-17 02:27:40.000000000 +0200
+++ glib-2.56.2/Makefile.in	2018-08-29 17:44:36.654734705 +0200
@@ -834,7 +834,7 @@
 
 @ENABLE_INSTALLED_TESTS_TRUE@installed_test_meta_DATA = $(installed_testcases:=.test)
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
-SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs tests
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
 DIST_SUBDIRS = $(SUBDIRS) build win32
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
