#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur gnutls-2.12.23.orig/Makefile.in gnutls-2.12.23/Makefile.in
--- gnutls-2.12.23.orig/Makefile.in	2015-05-25 19:12:24.598287603 +0200
+++ gnutls-2.12.23/Makefile.in	2015-05-25 19:12:45.722858220 +0200
@@ -205,7 +205,7 @@
  distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = lib libextra gl src doc tests guile
+DIST_SUBDIRS = lib libextra gl src tests guile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -1144,7 +1144,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
-SUBDIRS = lib libextra gl src doc tests $(am__append_1)
+SUBDIRS = lib libextra gl src tests $(am__append_1)
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/gl/m4 -I libextra/gl/m4 -I lib/m4 -I libextra/m4
 EXTRA_DIST = cfg.mk maint.mk .clcopying
