#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN claws-mail-3.11.1git179/src/folderview.c ../src/playground/build/claws-mail-3.11.1git179/src/folderview.c
--- claws-mail-3.11.1git179/src/folderview.c	2015-06-14 13:34:14.000000000 +0200
+++ ../src/playground/build/claws-mail-3.11.1git179/src/folderview.c	2015-07-04 18:14:57.245474029 +0200
@@ -876,12 +876,14 @@
 	if (folderview->open_folder) {
 		return;
 	}
-
+	
+	gtk_cmclist_freeze(GTK_CMCLIST(ctree));
 	gtkut_ctree_expand_parent_all(ctree, node);
 
 	folderview->open_folder = TRUE;
 	gtkut_ctree_set_focus_row(ctree, node);
 	gtk_cmctree_select(ctree, node);
+	gtk_cmclist_thaw(GTK_CMCLIST(ctree));
 	if ((folderview->summaryview->folder_item &&
 	    folderview->summaryview->folder_item->total_msgs > 0) ||
 	     prefs_common.layout_mode == SMALL_LAYOUT)
@@ -2496,21 +2498,24 @@
 
 	statusbar_verbosity_set(FALSE);
 	folder_item_update_freeze();
+	gtk_cmclist_freeze(GTK_CMCLIST(folderview->ctree));
 	if ((status = folder_item_move_to(from_folder, to_folder, &new_folder, copy)) == F_MOVE_OK) {
 		statusbar_verbosity_set(FALSE);
 		main_window_cursor_normal(folderview->mainwin);
 		STATUSBAR_POP(folderview->mainwin);
-		folder_item_update_thaw();
 		folder_item_update_recursive(new_folder, F_ITEM_UPDATE_MSGCNT);
 
 		folderview_sort_folders(folderview, 
 			gtk_cmctree_find_by_row_data(GTK_CMCTREE(folderview->ctree), 
 				NULL, to_folder), new_folder->folder);
+		gtk_cmclist_thaw(GTK_CMCLIST(folderview->ctree));
+		folder_item_update_thaw();
 		folderview_select(folderview, new_folder);
 	} else {
 		statusbar_verbosity_set(FALSE);		
 		main_window_cursor_normal(folderview->mainwin);
 		STATUSBAR_POP(folderview->mainwin);
+		gtk_cmclist_thaw(GTK_CMCLIST(folderview->ctree));
 		folder_item_update_thaw();
 		switch (status) {
 		case F_MOVE_FAILED_DEST_IS_PARENT:
Binary files claws-mail-3.11.1git179/src/.folderview.c.swp and ../src/playground/build/claws-mail-3.11.1git179/src/.folderview.c.swp differ
