#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN pixman-0.32.6-orig/Makefile.in pixman-0.32.6/Makefile.in
--- pixman-0.32.6-orig/Makefile.in	2014-07-06 00:54:03.000000000 +0200
+++ pixman-0.32.6/Makefile.in	2018-07-24 20:21:33.775036579 +0200
@@ -369,7 +369,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = pixman demos test
+SUBDIRS = pixman
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pixman-1.pc
 GPGKEY = 3892336E
