#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN cyrus-sasl-2.1.26-orig/plugins/Makefile.am cyrus-sasl-2.1.26/plugins/Makefile.am
--- cyrus-sasl-2.1.26-orig/plugins/Makefile.am	2016-05-29 18:49:02.580650805 +0200
+++ cyrus-sasl-2.1.26/plugins/Makefile.am	2016-05-29 18:57:24.763514573 +0200
@@ -53,6 +53,7 @@
 AM_LDFLAGS = -module -export-dynamic -rpath $(plugindir) -version-info $(plugin_version)
 
 COMPAT_OBJS = @LTGETADDRINFOOBJS@ @LTGETNAMEINFOOBJS@ @LTSNPRINTFOBJS@
+COMPAT_OBJS_REAL = @GETADDRINFOOBJS@ @GETNAMEINFOOBJS@ @SNPRINTFOBJS@
 
 EXTRA_DIST = makeinit.sh NTMakefile
 noinst_SCRIPTS = makeinit.sh
@@ -154,5 +155,7 @@
 	rm -f $(COMPAT_OBJS)
 	cd ../lib; $(MAKE) $(COMPAT_OBJS)
 	for file in $(COMPAT_OBJS); do ln -s ../lib/$$file .; done
+	-[ -d .libs] || mkdir .libs
+	for file in $(COMPAT_OBJS_REAL); do ln -s ../../lib/.libs/$$file .libs/; done
 
 
