#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN gnutls-3.4.7-orig/src/socket.c gnutls-3.4.7/src/socket.c
--- gnutls-3.4.7-orig/src/socket.c	2015-12-08 20:40:35.375075591 +0100
+++ gnutls-3.4.7/src/socket.c	2015-12-08 20:40:50.227457123 +0100
@@ -263,7 +263,6 @@
 	if (s != NULL) {
 		return s->s_port;
 	}
-	endservent();
 
 	return 443;
 }
@@ -278,7 +277,6 @@
 	if (s != NULL) {
 		return s->s_name;
 	}
-	endservent();
 
 	return "443";
 }
