#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN libetpan-1.7-orig/Makefile.in libetpan-1.7/Makefile.in
--- libetpan-1.7-orig/Makefile.in	2016-11-24 21:35:35.000000000 +0100
+++ libetpan-1.7/Makefile.in	2016-11-24 21:43:36.376698379 +0100
@@ -501,7 +501,7 @@
 # rules defined in rules.mk.
 CLEANFILES = stamp-prepare libetpan-config.h 
 DISTCLEANFILES = libetpan-config
-SUBDIRS = build-windows include src tests doc
+SUBDIRS = build-windows include src doc
 etpaninclude_HEADERS = libetpan-config.h
 all: $(BUILT_SOURCES) config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
