#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN libetpan-1.7-orig/src/data-types/mailsasl.c libetpan-1.7/src/data-types/mailsasl.c
--- libetpan-1.7-orig/src/data-types/mailsasl.c	2017-04-21 15:32:20.432729431 +0200
+++ libetpan-1.7/src/data-types/mailsasl.c	2017-04-21 15:33:24.015481671 +0200
@@ -126,6 +126,13 @@
   UNLOCK_SASL();
 }
 
+void mailsasl_set_plugin_path(char *path)
+{
+	LOCK_SASL();
+	sasl_set_path(SASL_PATH_TYPE_PLUGIN, path);
+	UNLOCK_SASL();
+}
+
 #else
 
 void mailsasl_external_ref(void)
@@ -140,4 +147,8 @@
 {
 }
 
+void mailsasl_set_plugin_path(char *path)
+{
+}
+
 #endif
diff -ruN libetpan-1.7-orig/src/data-types/mailsasl.h libetpan-1.7/src/data-types/mailsasl.h
--- libetpan-1.7-orig/src/data-types/mailsasl.h	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/data-types/mailsasl.h	2017-04-21 15:32:33.080481220 +0200
@@ -41,6 +41,7 @@
 
 void mailsasl_ref(void);
 void mailsasl_unref(void);
+void mailsasl_set_plugin_path(char *path);
 
 #ifdef __cplusplus
 }
diff -ruN libetpan-1.7-orig/src/low-level/imap/mailimap_socket.c libetpan-1.7/src/low-level/imap/mailimap_socket.c
--- libetpan-1.7-orig/src/low-level/imap/mailimap_socket.c	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/low-level/imap/mailimap_socket.c	2017-04-21 15:32:33.080481220 +0200
@@ -53,6 +53,10 @@
 
 #include "mailstream_cfstream.h"
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_IMAP_PORT 143
 #define SERVICE_NAME_IMAP "imap2"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/imap/mailimap_ssl.c libetpan-1.7/src/low-level/imap/mailimap_ssl.c
--- libetpan-1.7-orig/src/low-level/imap/mailimap_ssl.c	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/low-level/imap/mailimap_ssl.c	2017-04-21 15:32:33.080481220 +0200
@@ -53,6 +53,10 @@
 
 #include "mailstream_cfstream.h"
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_IMAPS_PORT 993
 #define SERVICE_NAME_IMAPS "imaps"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/nntp/newsnntp_socket.c libetpan-1.7/src/low-level/nntp/newsnntp_socket.c
--- libetpan-1.7-orig/src/low-level/nntp/newsnntp_socket.c	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/low-level/nntp/newsnntp_socket.c	2017-04-21 15:32:33.080481220 +0200
@@ -51,6 +51,10 @@
 #	include <unistd.h>
 #endif
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_NNTP_PORT 119
 #define SERVICE_NAME_NNTP "nntp"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/nntp/newsnntp_ssl.c libetpan-1.7/src/low-level/nntp/newsnntp_ssl.c
--- libetpan-1.7-orig/src/low-level/nntp/newsnntp_ssl.c	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/low-level/nntp/newsnntp_ssl.c	2017-04-21 15:32:33.080481220 +0200
@@ -50,6 +50,10 @@
 #	include <unistd.h>
 #endif
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_NNTPS_PORT 563
 #define SERVICE_NAME_NNTPS "nntps"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/pop3/mailpop3_socket.c libetpan-1.7/src/low-level/pop3/mailpop3_socket.c
--- libetpan-1.7-orig/src/low-level/pop3/mailpop3_socket.c	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/low-level/pop3/mailpop3_socket.c	2017-04-21 15:32:33.080481220 +0200
@@ -51,6 +51,10 @@
 #endif
 #include <stdlib.h>
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_POP3_PORT 110
 #define SERVICE_NAME_POP3 "pop3"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/pop3/mailpop3_ssl.c libetpan-1.7/src/low-level/pop3/mailpop3_ssl.c
--- libetpan-1.7-orig/src/low-level/pop3/mailpop3_ssl.c	2017-04-21 15:32:20.436729353 +0200
+++ libetpan-1.7/src/low-level/pop3/mailpop3_ssl.c	2017-04-21 15:32:33.080481220 +0200
@@ -51,6 +51,10 @@
 #endif
 #include <stdlib.h>
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_POP3S_PORT 995
 #define SERVICE_NAME_POP3S "pop3s"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/smtp/mailsmtp_socket.c libetpan-1.7/src/low-level/smtp/mailsmtp_socket.c
--- libetpan-1.7-orig/src/low-level/smtp/mailsmtp_socket.c	2017-04-21 15:32:20.440729274 +0200
+++ libetpan-1.7/src/low-level/smtp/mailsmtp_socket.c	2017-04-21 15:32:33.080481220 +0200
@@ -52,6 +52,10 @@
 #	include <unistd.h>
 #endif
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_SMTP_PORT 25
 #define SERVICE_NAME_SMTP "smtp"
 #define SERVICE_TYPE_TCP "tcp"
diff -ruN libetpan-1.7-orig/src/low-level/smtp/mailsmtp_ssl.c libetpan-1.7/src/low-level/smtp/mailsmtp_ssl.c
--- libetpan-1.7-orig/src/low-level/smtp/mailsmtp_ssl.c	2017-04-21 15:32:20.440729274 +0200
+++ libetpan-1.7/src/low-level/smtp/mailsmtp_ssl.c	2017-04-21 15:32:33.080481220 +0200
@@ -52,6 +52,10 @@
 #	include <unistd.h>
 #endif
 
+#ifdef HAVE_MINGW32_SYSTEM
+# include <winsock2.h>
+#endif
+
 #define DEFAULT_SMTPS_PORT 465
 #define SERVICE_NAME_SMTPS "smtps"
 #define SERVICE_TYPE_TCP "tcp"
