#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN claws-mail-3.13.2git142-orig/src/main.c claws-mail-3.13.2git142/src/main.c
--- claws-mail-3.13.2git142-orig/src/main.c	2016-05-29 18:22:35.230016702 +0200
+++ claws-mail-3.13.2git142/src/main.c	2016-05-29 18:28:15.729204613 +0200
@@ -131,7 +131,9 @@
 #ifdef HAVE_LIBETPAN
 #include "imap-thread.h"
 #include "nntp-thread.h"
+extern void mailsasl_set_plugin_path(char *path);
 #endif
+
 #include "stock_pixmap.h"
 #ifdef USE_GNUTLS
 #  include "ssl.h"
@@ -1307,6 +1309,13 @@
 	account_read_config_all();
 
 #ifdef HAVE_LIBETPAN
+	/* Following block utilizes unofficial libetpan API, added
+	 * downstream in our claws-win32-installer repository. */
+	gchar *pluginpath = g_strdup_printf("%s\\lib\\sasl2",
+			claws_get_startup_dir());
+	mailsasl_set_plugin_path(pluginpath);
+	g_free(pluginpath);
+
 	imap_main_init(prefs_common.skip_ssl_cert_check);
 	imap_main_set_timeout(prefs_common.io_timeout_secs);
 	nntp_main_init(prefs_common.skip_ssl_cert_check);
