#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/ldaputil.h claws-mail-3.11.1/src/ldaputil.h
--- claws-mail-3.11.1-orig/src/ldaputil.h	2015-06-18 15:10:11.371528056 +0200
+++ claws-mail-3.11.1/src/ldaputil.h	2015-06-20 22:16:10.159963428 +0200
@@ -37,6 +37,9 @@
 #define ldap_unbind_ext(ld,x,y) ldap_unbind_s(ld)
 #define LDAP_ADMINLIMIT_EXCEEDED LDAP_ADMIN_LIMIT_EXCEEDED
 #define timeval l_timeval
+#ifndef LDAP_OPT_SUCCESS
+# define LDAP_OPT_SUCCESS 0
+#endif
 #endif
 /* Function Prototypes */
 GList *ldaputil_read_basedn	( const gchar *host, const gint port,
