#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/claws.def claws-mail-3.11.1/src/plugins/rssyl/claws.def
--- claws-mail-3.11.1-orig/src/plugins/rssyl/claws.def	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/claws.def	2015-06-18 23:18:59.118874089 +0200
@@ -1,13 +1,13 @@
 LIBRARY CLAWS-MAIL.EXE
 EXPORTS
-get_locale_dir
-check_plugin_version
 alertpanel_error
 alertpanel_full
 change_dir
+check_plugin_version
 claws_do_idle
 claws_is_exiting
 claws_is_starting
+claws_ssl_get_cert_file
 claws_unlink
 cm_menu_set_sensitive_full
 conv_encode_header_full
@@ -32,13 +32,19 @@
 folder_item_append
 folder_item_get_identifier
 folder_item_get_msginfo
+folder_item_get_name
 folder_item_get_path
+folder_item_get_xml
 folder_item_new
 folder_item_parent
+folder_item_prefs_copy_prefs
+folder_item_prefs_save_config
 folder_item_prefs_save_config_recursive
 folder_item_remove
 folder_item_rename
 folder_item_scan
+folder_item_search_msgs_local
+folder_item_set_xml
 folder_item_update_freeze
 folder_item_update_thaw
 folder_local_folder_destroy
@@ -53,18 +59,23 @@
 folderview_unregister_popup
 folderview_unselect
 folder_write_list
+get_locale_dir
 get_rc_dir
 get_tmp_file
+gtk_cmclist_freeze
+gtk_cmclist_get_type
+gtk_cmclist_thaw
 gtk_cmctree_get_type
 gtk_cmctree_is_ancestor
 gtk_cmctree_node_get_row_data
 gtkut_widget_draw_now
-gtkut_window_new
 gtkut_widget_set_can_default
+gtkut_window_new
 inc_offline_should_override
 input_dialog
 is_dir_exist
 is_file_entry_exist
+itos
 log_error
 log_print
 log_warning
@@ -73,13 +84,17 @@
 mainwindow_get_mainwindow
 make_dir
 make_dir_hier
-mkstemp
+mh_get_class
+move_file
+msgcache_destroy
+prefs_common
 prefs_common_get_prefs
 prefs_file_close
 prefs_file_close_revert
 prefs_filtering_delete_path
 prefs_gtk_register_page
 prefs_gtk_unregister_page
+prefs_matcher_write_config
 prefs_read_config
 prefs_set_block_label
 prefs_set_default
@@ -91,9 +106,13 @@
 procheader_parse_file
 procmsg_get_message_file
 procmsg_msginfo_unset_flags
+remove_dir_recursive
+slist_free_strings_full
 strtailchomp
 subst_for_shellsafe_filename
 summary_clear_all
 summary_show
 to_number
 trim_string
+xml_attr_new
+xml_tag_add_attr
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/date.c claws-mail-3.11.1/src/plugins/rssyl/libfeed/date.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/date.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/date.c	2015-06-18 22:19:51.975709663 +0200
@@ -37,78 +37,6 @@
 
 #include <time.h>
 #include <glib.h>
-#include <locale.h>
-#include <string.h>
-#include <ctype.h>
-#include <stdlib.h>
-
-/* converts a ISO 8601 time string to a time_t value */
-time_t parseISO8601Date(gchar *date) {
-	struct tm	tm;
-	time_t		t, t2, offset = 0;
-	gboolean	success = FALSE;
-	gchar *pos;
-
-	if (date == NULL)
-		return -1;
-	
-	memset(&tm, 0, sizeof(struct tm));
-	
-	/* we expect at least something like "2003-08-07T15:28:19" and
-	   don't require the second fractions and the timezone info
-
-	   the most specific format:   YYYY-MM-DDThh:mm:ss.sTZD
-	 */
-	 
-	/* full specified variant */
-	if(NULL != (pos = strptime((const char *)date, "%Y-%m-%dT%H:%M:%SZ", &tm))) {
-		/* Parse seconds */
-		if (*pos == ':')
-			pos++;
-		if (isdigit(pos[0]) && !isdigit(pos[1])) {
-			tm.tm_sec = pos[0] - '0';
-			pos++;
-		} else if (isdigit(pos[0]) && isdigit(pos[1])) {
-			tm.tm_sec = 10*(pos[0]-'0') + pos[1] - '0';
-			pos +=2;
-		}
-		/* Parse timezone */
-		if (*pos == 'Z')
-			offset = 0;
-		else if ((*pos == '+' || *pos == '-') && isdigit(pos[1]) && isdigit(pos[2]) && strlen(pos) >= 3) {
-			offset = (10*(pos[1] - '0') + (pos[2] - '0')) * 60 * 60;
-			
-			if (pos[3] == ':' && isdigit(pos[4]) && isdigit(pos[5]))
-				offset +=  (10*(pos[4] - '0') + (pos[5] - '0')) * 60;
-			else if (isdigit(pos[3]) && isdigit(pos[4]))
-				offset +=  (10*(pos[3] - '0') + (pos[4] - '0')) * 60;
-			
-			offset *= (pos[0] == '+') ? 1 : -1;
-
-		}
-		success = TRUE;
-	/* only date */
-	} else if(NULL != strptime((const char *)date, "%t%Y-%m-%d", &tm))
-		success = TRUE;
-	/* there were others combinations too... */
-
-	if(TRUE == success) {
-		if((time_t)(-1) != (t = mktime(&tm))) {
-			/* Correct for the local timezone*/
-			t = t - offset;
-			t2 = mktime(gmtime(&t));
-			t = t - (t2 - t);
-			
-			return t;
-		} else {
-			g_warning("internal error! time conversion error! mktime failed!\n");
-		}
-	} else {
-		g_warning("Invalid ISO8601 date format! Ignoring <dc:date> information!\n");
-	}
-	
-	return 0;
-}
 
 gchar *dayofweek[] = { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
 gchar *months[] = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
@@ -120,29 +48,3 @@
 	return g_strdup_printf("%s, %2d %s %4d %02d:%02d:%02d GMT", dayofweek[tm->tm_wday], tm->tm_mday,
 					   months[tm->tm_mon], 1900 + tm->tm_year, tm->tm_hour, tm->tm_min, tm->tm_sec);
 }
-
-time_t parseRFC822Date(gchar *date)
-{
-	struct tm t;
-	memset(&t, 0, sizeof(struct tm));
-	const char *c = setlocale(LC_TIME, NULL);
-
-	/* Adjust the LC_TIME locale to standard C in order for strptime()
-	 * to work reliably. */
-	if (c != NULL)
-		setlocale(LC_TIME, "C");
-
-	if (!strptime(date, "%a, %d %b %Y %H:%M:%S %Z", &t) &&
-		!strptime(date, "%a, %d %b %Y %H:%M %Z", &t)) {
-		g_warning("Invalid RFC822 date!\n");
-		if (c != NULL)
-			setlocale(LC_TIME, c);
-		return 0;
-	}
-
-	/* Restore the original LC_TIME locale. */
-	if (c != NULL)
-		setlocale(LC_TIME, c);
-
-	return mktime(&t);
-}
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/date.h claws-mail-3.11.1/src/plugins/rssyl/libfeed/date.h
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/date.h	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/date.h	2015-06-18 22:19:51.975709663 +0200
@@ -4,8 +4,6 @@
 #include <time.h>
 #include <glib.h>
 
-time_t parseISO8601Date(gchar *date);
 gchar *createRFC822Date(const time_t *time);
-time_t parseRFC822Date(gchar *date);
 
 #endif /* __DATE_H */
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/feed.c claws-mail-3.11.1/src/plugins/rssyl/libfeed/feed.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/feed.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/feed.c	2015-06-18 23:14:08.015082287 +0200
@@ -267,6 +267,9 @@
 	}
 #endif
 
+	if(feed->cacert_file != NULL)
+		curl_easy_setopt(eh, CURLOPT_CAINFO, feed->cacert_file);
+
 	if(feed->cookies_path != NULL)
 		curl_easy_setopt(eh, CURLOPT_COOKIEFILE, feed->cookies_path);
 
@@ -355,3 +358,21 @@
 	g_return_if_fail(feed != NULL);
 	feed->ssl_verify_peer = ssl_verify_peer;
 }
+
+gchar *feed_get_cacert_file(Feed *feed)
+{
+	g_return_val_if_fail(feed != NULL, NULL);
+	return feed->cacert_file;
+}
+
+void feed_set_cacert_file(Feed *feed, const gchar *path)
+{
+	g_return_if_fail(feed != NULL);
+
+	if( feed->cacert_file != NULL ) {
+		g_free(feed->cacert_file);
+		feed->cacert_file = NULL;
+	}
+
+	feed->cacert_file = (path != NULL ? g_strdup(path) : NULL);
+}
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/feed.h claws-mail-3.11.1/src/plugins/rssyl/libfeed/feed.h
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/feed.h	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/feed.h	2015-06-18 23:07:39.336684751 +0200
@@ -44,6 +44,7 @@
 	gchar *fetcherr;
 	gchar *cookies_path;
 	gboolean ssl_verify_peer;
+	gchar *cacert_file;
 
 	GSList *items;
 };
@@ -96,6 +97,9 @@
 gboolean feed_get_ssl_verify_peer(Feed *feed);
 void feed_set_ssl_verify_peer(Feed *feed, gboolean ssl_verify_peer);
 
+gchar *feed_get_cacert_file(Feed *feed);
+void feed_set_cacert_file(Feed *feed, const gchar *path);
+
 gint feed_n_items(Feed *feed);
 FeedItem *feed_nth_item(Feed *feed, guint n);
 
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/parser_atom10.c claws-mail-3.11.1/src/plugins/rssyl/libfeed/parser_atom10.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/parser_atom10.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/parser_atom10.c	2015-06-18 22:19:51.975709663 +0200
@@ -23,6 +23,8 @@
 #include <string.h>
 #include <stdio.h>
 
+#include <procheader.h>
+
 #include "feed.h"
 #include "feeditem.h"
 #include "date.h"
@@ -141,7 +143,7 @@
 			} else if( !strcmp(el, "summary" ) ) {
 				FILL(feed->description)
 			} else if( !strcmp(el, "updated" ) ) {
-				feed->date = parseISO8601Date(text);
+				feed->date = procheader_date_parse(NULL, text, 0);
 			}
 			/* FIXME: add more later */
 
@@ -167,9 +169,9 @@
 						FILL(ctx->curitem->id)
 						feed_item_set_id_permalink(ctx->curitem, TRUE);
 					} else if( !strcmp(el, "published") ) {
-						ctx->curitem->date_published = parseISO8601Date(text);
+						ctx->curitem->date_published = procheader_date_parse(NULL, text, 0);
 					} else if( !strcmp(el, "updated") ) {
-						ctx->curitem->date_modified = parseISO8601Date(text);
+						ctx->curitem->date_modified = procheader_date_parse(NULL, text, 0);
 					}
 
 					break;
@@ -218,7 +220,7 @@
 					} else if( !strcmp(el, "id" ) ) {
 						FILL(ctx->curitem->sourceid)
 					} else if( !strcmp(el, "updated" ) ) {
-						ctx->curitem->sourcedate = parseISO8601Date(text);
+						ctx->curitem->sourcedate = procheader_date_parse(NULL, text, 0);
 					}
 
 					break;
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/parser_rdf.c claws-mail-3.11.1/src/plugins/rssyl/libfeed/parser_rdf.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/parser_rdf.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/parser_rdf.c	2015-06-18 22:19:51.975709663 +0200
@@ -22,6 +22,8 @@
 #include <expat.h>
 #include <string.h>
 
+#include <procheader.h>
+
 #include "feed.h"
 #include "date.h"
 #include "parser_rdf.h"
@@ -102,9 +104,9 @@
 					} else if( !strcmp(el, "dc:creator") ) {
 						FILL(feed->author)
 					} else if( !strcmp(el, "dc:date") ) {
-						feed->date = parseISO8601Date(text);
+						feed->date = procheader_date_parse(NULL, text, 0);
 					} else if( !strcmp(el, "pubDate") ) {
-						feed->date = parseRFC822Date(text);
+						feed->date = procheader_date_parse(NULL, text, 0);
 					}
 
 					break;
@@ -127,9 +129,9 @@
 					} else if( !strcmp(el, "link") ) {
 						FILL(ctx->curitem->url)
 					} else if( !strcmp(el, "dc:date") ) {
-						ctx->curitem->date_modified = parseISO8601Date(text);
+						ctx->curitem->date_modified = procheader_date_parse(NULL, text, 0);
 					} else if( !strcmp(el, "pubDate") ) {
-						ctx->curitem->date_modified = parseRFC822Date(text);
+						ctx->curitem->date_modified = procheader_date_parse(NULL, text, 0);
 					}
 
 					break;
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/parser_rss20.c claws-mail-3.11.1/src/plugins/rssyl/libfeed/parser_rss20.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/libfeed/parser_rss20.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/libfeed/parser_rss20.c	2015-06-18 22:19:51.975709663 +0200
@@ -22,6 +22,8 @@
 #include <expat.h>
 #include <string.h>
 
+#include <procheader.h>
+
 #include "feed.h"
 #include "feeditem.h"
 #include "feeditemenclosure.h"
@@ -130,9 +132,9 @@
 			} else if( !strcmp(el, "admin:generatorAgent") ) {
 				FILL(feed->generator)
 			} else if( !strcmp(el, "dc:date") ) {
-				feed->date = parseISO8601Date(text);
+				feed->date = procheader_date_parse(NULL, text, 0);
 			} else if( !strcmp(el, "pubDate") ) {
-				feed->date = parseRFC822Date(text);
+				feed->date = procheader_date_parse(NULL, text, 0);
 			}
 
 			break;
@@ -160,9 +162,9 @@
 			} else if( !strcmp(el, "wfw:commentRSS") || !strcmp(el, "wfw:commentRss") ) {
 				FILL(ctx->curitem->comments_url)
 			} else if( !strcmp(el, "dc:date") ) {
-				ctx->curitem->date_modified = parseISO8601Date(text);
+				ctx->curitem->date_modified = procheader_date_parse(NULL, text, 0);
 			} else if( !strcmp(el, "pubDate") ) {
-				ctx->curitem->date_modified = parseRFC822Date(text);
+				ctx->curitem->date_modified = procheader_date_parse(NULL, text, 0);
 			} else if( !strcmp(el, "dc:creator")) {
 				FILL(ctx->curitem->author)
 			}
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/parse822.c claws-mail-3.11.1/src/plugins/rssyl/parse822.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/parse822.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/parse822.c	2015-06-18 22:19:51.979709771 +0200
@@ -23,7 +23,6 @@
 #endif
 
 /* Global includes */
-#include <sys/stat.h>
 #include <glib.h>
 #include <pthread.h>
 
@@ -241,12 +240,12 @@
 static void rssyl_folder_read_existing_real(RFolderItem *ritem)
 {
 	gchar *path = NULL, *fname = NULL;
-	DIR *dp;
-	struct dirent *d;
-	struct stat st;
+	GDir *dp;
+	const gchar *d;
 	gint num;
 	FeedItem *item = NULL;
 	RFeedCtx *ctx;
+	GError *error;
 
 	g_return_if_fail(ritem != NULL);
 
@@ -263,34 +262,29 @@
 	ritem->items = NULL;
 	ritem->last_update = 0;
 
-	if( (dp = opendir(path)) == NULL ) {
-		FILE_OP_ERROR(path, "opendir");
+	if( (dp = g_dir_open(path, 0, &error)) == NULL ) {
+		FILE_OP_ERROR(path, "g_dir_open");
+		debug_print("g_dir_open on \"%s\" failed with error %d (%s)\n",
+				path, error->code, error->message);
 		g_free(path);
 		return;
 	}
 
-	while( (d = readdir(dp)) != NULL ) {
+	while( (d = g_dir_read_name(dp)) != NULL ) {
 		if( claws_is_exiting() ) {
-			closedir(dp);
-			g_free(path);
+			g_dir_close(dp);
 			return;
 		}
 
-		if( d->d_name[0] != '.' && (num = to_number(d->d_name)) > 0 ) {
-			fname = g_strdup_printf("%s%c%s", path, G_DIR_SEPARATOR, d->d_name);
-			if( g_stat(fname, &st) < 0 ) {
-				debug_print("RSSyl: couldn't stat() file '%s', ignoring it\n", fname);
-				g_free(fname);
-				continue;
-			}
-
-			if( !S_ISREG(st.st_mode)) {
+		if( d[0] != '.' && (num = to_number(d)) > 0 ) {
+			fname = g_strdup_printf("%s%c%s", path, G_DIR_SEPARATOR, d);
+			if (!g_file_test(fname, G_FILE_TEST_IS_REGULAR)) {
 				debug_print("RSSyl: not a regular file: '%s', ignoring it\n", fname);
 				g_free(fname);
 				continue;
 			}
 
-			debug_print("RSSyl: starting to parse '%s'\n", d->d_name);
+			debug_print("RSSyl: starting to parse '%s'\n", d);
 			if( (item = rssyl_parse_folder_item_file(fname)) != NULL ) {
 				/* Find latest timestamp */
 				ctx = (RFeedCtx *)item->data;
@@ -303,7 +297,7 @@
 		}
 	}
 
-	closedir(dp);
+	g_dir_close(dp);
 	g_free(path);
 
 	ritem->items = g_slist_reverse(ritem->items);
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl_add_item.c claws-mail-3.11.1/src/plugins/rssyl/rssyl_add_item.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl_add_item.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/rssyl_add_item.c	2015-06-18 22:19:51.979709771 +0200
@@ -374,7 +374,7 @@
 	dirname = folder_item_get_path(&ritem->item);
 	template = g_strconcat(dirname, G_DIR_SEPARATOR_S,
 			RSSYL_TMP_TEMPLATE, NULL);
-	if ((fd = mkstemp(template)) < 0) {
+	if ((fd = g_mkstemp(template)) < 0) {
 		g_warning("Couldn't mkstemp('%s'), not adding message!\n", template);
 		g_free(template);
 		return;
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl.c claws-mail-3.11.1/src/plugins/rssyl/rssyl.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/rssyl.c	2015-06-18 22:19:51.979709771 +0200
@@ -203,8 +203,10 @@
 static void rssyl_get_last_num(Folder *folder, FolderItem *item)
 {
 	gchar *path;
-	DIR *dp;
-	struct dirent *d;
+	const gchar *f;
+	GDir *dp;
+	GError *gerror;
+
 	gint max = 0;
 	gint num;
 
@@ -214,21 +216,24 @@
 	path = folder_item_get_path(item);
 	g_return_if_fail(path != NULL);
 
-	if( (dp = opendir(path)) == NULL ) {
-		FILE_OP_ERROR(item->path, "opendir");
+	if( (dp = g_dir_open(path, 0, &gerror)) == NULL ) {
+		debug_print("g_dir_open() failed on \"%s\", error %d (%s).\n",
+				path, gerror->code, gerror->message);
+		FILE_OP_ERROR(item->path, "g_dir_open");
 		g_free(path);
 		return;
 	}
 
 	g_free(path);
 
-	while( (d = readdir(dp)) != NULL ) {
-		if( (num = to_number(d->d_name)) > 0 && dirent_is_regular_file(d) ) {
+	while( (f = g_dir_read_name(dp)) != NULL) {
+		if ((num = to_number(f)) > 0 &&
+				g_file_test(f, G_FILE_TEST_IS_REGULAR)) {
 			if( max < num )
 				max = num;
 		}
 	}
-	closedir(dp);
+	g_dir_close(dp);
 
 	debug_print("Last number in dir %s = %d\n", item->path, max);
 	item->last_num = max;
@@ -489,7 +494,7 @@
 
 	name = folder_item_get_name(rssyl_get_root_folderitem(item));
 	path = g_strconcat(get_rc_dir(), G_DIR_SEPARATOR_S, RSSYL_DIR,
-			G_DIR_SEPARATOR_S, name, G_DIR_SEPARATOR_S, item->path, NULL);
+			G_DIR_SEPARATOR_S, name, item->path, NULL);
 	g_free(name);
 
 	return path;
@@ -612,8 +617,9 @@
 		MsgNumberList **list, gboolean *old_uids_valid)
 {
 	gchar *path;
-	DIR *dp;
-	struct dirent *d;
+	GDir *dp;
+	const gchar *d;
+	GError *error;
 	gint num, nummsgs = 0;
 
 	g_return_val_if_fail(item != NULL, -1);
@@ -625,22 +631,23 @@
 	path = folder_item_get_path(item);
 	g_return_val_if_fail(path != NULL, -1);
 
-	if( (dp = opendir(path)) == NULL ) {
-		FILE_OP_ERROR(item->path, "opendir");
+	if( (dp = g_dir_open(path, 0, &error)) == NULL ) {
+		FILE_OP_ERROR(item->path, "g_dir_open");
+		debug_print("g_dir_open() failed on \"%s\", error %d (%s).\n",
+				path, error->code, error->message);
 		g_free(path);
 		return -1;
 	}
 
 	g_free(path);
 
-	while( (d = readdir(dp)) != NULL ) {
-		if( (num = to_number(d->d_name)) > 0 ) {
+	while( (d = g_dir_read_name(dp)) != NULL ) {
+		if( (num = to_number(d)) > 0 ) {
 			*list = g_slist_prepend(*list, GINT_TO_POINTER(num));
 			nummsgs++;
 		}
 	}
-
-	closedir(dp);
+	g_dir_close(dp);
 
 	debug_print("RSSyl: get_num_list: returning %d\n", nummsgs);
 
@@ -650,7 +657,7 @@
 static gboolean rssyl_is_msg_changed(Folder *folder, FolderItem *item,
 		MsgInfo *msginfo)
 {
-	struct stat s;
+	GStatBuf s;
 	gchar *path = NULL;
 
 	g_return_val_if_fail(folder != NULL, FALSE);
@@ -744,12 +751,10 @@
 		g_return_val_if_fail(destfile != NULL, -1);
 		debug_print("RSSyl: add_msgs: new filename is '%s'\n", destfile);
 
-		if( link(fileinfo->file, destfile) < 0 ) {
-			if( copy_file(fileinfo->file, destfile, TRUE) < 0 ) {
-				g_warning("can't copy message %s to %s\n", fileinfo->file, destfile);
-				g_free(destfile);
-				return -1;
-			}
+		if( copy_file(fileinfo->file, destfile, TRUE) < 0) {
+			g_warning("can't copy message %s to %s\n", fileinfo->file, destfile);
+			g_free(destfile);
+			return -1;
 		}
 
 		if( relation != NULL )
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl_update_comments.c claws-mail-3.11.1/src/plugins/rssyl/rssyl_update_comments.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl_update_comments.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/rssyl_update_comments.c	2015-06-18 22:19:51.979709771 +0200
@@ -56,8 +56,9 @@
 	FolderItem *item = &ritem->item;
 	FeedItem *fi = NULL;
 	RFetchCtx *ctx = NULL;
-	DIR *dp;
-	struct dirent *d;
+	GDir *dp;
+	const gchar *d;
+	GError *error;
 	gint num;
 	gchar *path, *msg, *fname;
 	MainWindow *mainwin = mainwindow_get_mainwindow();
@@ -72,26 +73,32 @@
 
 	debug_print("RSSyl: starting to parse comments, path is '%s'\n", path);
 
-	if( (dp = opendir(path)) == NULL ) {
-		FILE_OP_ERROR(item->path, "opendir");
+	if( (dp = g_dir_open(path, 0, &error)) == NULL ) {
+		FILE_OP_ERROR(item->path, "g_dir_open");
+		debug_print("g_dir_open on \"%s\" failed with error %d (%s)\n",
+				path, error->code, error->message);
+		g_error_free(error);
 		g_free(path);
 		return;
 	}
 
 	ritem->fetching_comments = TRUE;
 
-	while( (d = readdir(dp)) != NULL ) {
+	while( (d = g_dir_read_name(dp)) != NULL ) {
 		if (claws_is_exiting()) {
-			closedir(dp);
+			g_dir_close(dp);
 			g_free(path);
 			debug_print("RSSyl: bailing out, app is exiting\n");
 			return;
 		}
 
-		if( (num = to_number(d->d_name)) > 0 && d->d_type == DT_REG ) {
-			debug_print("RSSyl: starting to parse '%s'\n", d->d_name);
+		if( (num = to_number(d)) > 0) {
+			fname = g_strdup_printf("%s%c%s", path, G_DIR_SEPARATOR, d);
+			if (!g_file_test(fname, G_FILE_TEST_IS_REGULAR))
+				continue;
+
+			debug_print("RSSyl: starting to parse '%s'\n", d);
 
-			fname = g_strdup_printf("%s%c%s", path, G_DIR_SEPARATOR, d->d_name);
 			if( (fi = rssyl_parse_folder_item_file(fname)) != NULL ) {
 				if( feed_item_get_comments_url(fi) && feed_item_get_id(fi) &&
 						(ritem->fetch_comments_max_age == -1 ||
@@ -131,7 +138,7 @@
 		}
 	}
 
-	closedir(dp);
+	g_dir_close(dp);
 	g_free(path);
 
 	ritem->fetching_comments = FALSE;
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl_update_feed.c claws-mail-3.11.1/src/plugins/rssyl/rssyl_update_feed.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl_update_feed.c	2015-06-18 15:10:11.359527733 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/rssyl_update_feed.c	2015-06-18 23:44:58.012500049 +0200
@@ -165,6 +165,12 @@
 	feed_set_timeout(ctx->feed, prefs_common.io_timeout_secs);
 	feed_set_cookies_path(ctx->feed, rssyl_prefs_get()->cookies_path);
 	feed_set_ssl_verify_peer(ctx->feed, ritem->ssl_verify_peer);
+#ifdef G_OS_WIN32
+	if (!g_ascii_strncasecmp(ritem->url, "https", 5)) {
+		feed_set_cacert_file(ctx->feed, claws_ssl_get_cert_file());
+		debug_print("RSSyl: using cert file '%s'\n", feed_get_cacert_file(ctx->feed));
+	}
+#endif
 
 	return ctx;
 }
@@ -184,6 +190,12 @@
 	feed_set_timeout(ctx->feed, prefs_common.io_timeout_secs);
 	feed_set_cookies_path(ctx->feed, rssyl_prefs_get()->cookies_path);
 	feed_set_ssl_verify_peer(ctx->feed, rssyl_prefs_get()->ssl_verify_peer);
+#ifdef G_OS_WIN32
+	if (!g_ascii_strncasecmp(url, "https", 5)) {
+		feed_set_cacert_file(ctx->feed, claws_ssl_get_cert_file());
+		debug_print("RSSyl: using cert file '%s'\n", feed_get_cacert_file(ctx->feed));
+	}
+#endif
 
 	return ctx;
 }
diff -ur claws-mail-3.11.1-orig/src/procheader.c claws-mail-3.11.1/src/procheader.c
--- claws-mail-3.11.1-orig/src/procheader.c	2015-06-18 15:10:11.375528164 +0200
+++ claws-mail-3.11.1/src/procheader.c	2015-06-18 22:19:51.979709771 +0200
@@ -790,8 +790,10 @@
 {
 	gint result;
 	gint month_n;
-	gchar zone1[3];
-	gchar zone2[3];
+	gint secfract;
+	gint zone1, zone2;
+	gchar offset_sign;
+	gchar sep1;
 
 	if (str == NULL)
 		return -1;
@@ -836,23 +838,40 @@
 			day, month, year, hh, mm);
 	if (result == 5) return 0;
 
-	/* RFC3339 subset */
 	*weekday = '\0';
-	result = sscanf(str, "%4d-%2d-%2d %2d:%2d:%2d+%2s:%2s",
-			year, &month_n, day, hh, mm, ss, zone1, zone2);
-	if (result == 8) {
-		if (1 <= month_n && month_n <= 12) {
+
+	/* RFC3339 subset, with fraction of second */
+	/* TODO: Allow "Z" instead of time zone offset, since RFC allows
+	 * that too. */
+	result = sscanf(str, "%4d-%2d-%2d%c%2d:%2d:%2d.%1d%c%2d:%2d",
+			year, &month_n, day, &sep1, hh, mm, ss, &secfract,
+			&offset_sign, &zone1, &zone2);
+	if (result == 11
+			&& (sep1 == 'T' || sep1 == 't' || sep1 == ' ')) {
+		if (month_n >= 1 && month_n <= 12) {
 			strncpy2(month, monthstr+((month_n-1)*3), 4);
-			*zone = '+';
-			strncpy2(zone+1, zone1, 3);
-			strncpy2(zone+3, zone2, 3);
+			sprintf(zone, "%c%2d%2d", offset_sign, zone1, zone2);
+			return 0;
+		}
+	}
+
+	/* RFC3339 subset, no fraction of second */
+	result = sscanf(str, "%4d-%2d-%2d%c%2d:%2d:%2d%c%2d:%2d",
+			year, &month_n, day, &sep1, hh, mm, ss,
+			&offset_sign, &zone1, &zone2);
+	if (result == 10
+			&& (sep1 == 'T' || sep1 == 't' || sep1 == ' ')) {
+		if (month_n >= 1 && month_n <= 12) {
+			strncpy2(month, monthstr+((month_n-1)*3), 4);
+			sprintf(zone, "%c%2d%2d", offset_sign, zone1, zone2);
 			return 0;
 		}
 	}
 
-	/* RFC3339 subset */
 	*zone = '\0';
-	*weekday = '\0';
+
+	/* RFC3339 subset */
+	/* This particular "subset" is invalid, RFC requires the time offset */
 	result = sscanf(str, "%4d-%2d-%2d %2d:%2d:%2d",
 			year, &month_n, day, hh, mm, ss);
 	if (result == 6) {
