#! /bin/sh
patch -p0 -l -f $* < $0
exit $?

--- configure.orig	2015-06-02 15:02:52.934111437 +0200
+++ configure	2015-06-02 15:02:58.266255914 +0200
@@ -13910,7 +13910,7 @@
         # Note that we need to link to pthreadGC2 in all cases. This
         # is because some locking is used even when pthread support is
         # disabled.
-        pthread_name=pthreadGC2
+        pthread_name=pthread
 	CFLAGS="$CFLAGS -mms-bitfields"
 	LIBS="$LIBS -l${pthread_name} -lws2_32 -lregex"
 	;;
