#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/addrbook.c claws-mail-3.11.1/src/addrbook.c
--- claws-mail-3.11.1-orig/src/addrbook.c	2015-06-06 18:01:32.966681258 +0200
+++ claws-mail-3.11.1/src/addrbook.c	2015-06-06 18:02:19.707940582 +0200
@@ -1790,7 +1790,7 @@
 	gchar *adbookdir;
 	GDir *dir;
 	const gchar *dir_name;
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gchar buf[WORK_BUFLEN + 1];
 	gchar numbuf[WORK_BUFLEN];
 	gint len, lenpre, lensuf, lennum;
diff -ur claws-mail-3.11.1-orig/src/addrcache.c claws-mail-3.11.1/src/addrcache.c
--- claws-mail-3.11.1-orig/src/addrcache.c	2015-06-06 18:01:32.966681258 +0200
+++ claws-mail-3.11.1/src/addrcache.c	2015-06-06 18:02:31.420255938 +0200
@@ -242,7 +242,7 @@
 */
 gboolean addrcache_check_file( AddressCache *cache, gchar *path ) {
 	gboolean retVal;
-	struct stat filestat;
+	GStatBuf filestat;
 	retVal = TRUE;
 	if( path ) {
 		if( 0 == g_stat( path, &filestat ) ) {
@@ -258,7 +258,7 @@
 */
 gboolean addrcache_mark_file( AddressCache *cache, gchar *path ) {
 	gboolean retVal = FALSE;
-	struct stat filestat;
+	GStatBuf filestat;
 	if( path ) {
 		if( 0 == g_stat( path, &filestat ) ) {
 			cache->modifyTime = filestat.st_mtime;
diff -ur claws-mail-3.11.1-orig/src/addrharvest.c claws-mail-3.11.1/src/addrharvest.c
--- claws-mail-3.11.1-orig/src/addrharvest.c	2015-06-06 18:01:32.962681149 +0200
+++ claws-mail-3.11.1/src/addrharvest.c	2015-06-06 18:02:50.748775188 +0200
@@ -765,7 +765,7 @@
 {
 	DIR *dp;
 	struct dirent *d;
-	struct stat s;
+	GStatBuf s;
 	gint num;
 	int r;
 
diff -ur claws-mail-3.11.1-orig/src/common/template.c claws-mail-3.11.1/src/common/template.c
--- claws-mail-3.11.1-orig/src/common/template.c	2015-06-06 18:01:32.958681041 +0200
+++ claws-mail-3.11.1/src/common/template.c	2015-06-06 18:04:44.407828421 +0200
@@ -158,7 +158,7 @@
 	gchar *filename;
 	GDir *dir;
 	const gchar *dir_name;
-	struct stat s;
+	GStatBuf s;
 	Template *tmpl;
 	GSList *tmpl_list = NULL;
 
diff -ur claws-mail-3.11.1-orig/src/common/utils.c claws-mail-3.11.1/src/common/utils.c
--- claws-mail-3.11.1-orig/src/common/utils.c	2015-06-06 18:01:32.958681041 +0200
+++ claws-mail-3.11.1/src/common/utils.c	2015-06-06 18:04:21.399210351 +0200
@@ -2128,7 +2128,7 @@
 
 off_t get_file_size(const gchar *file)
 {
-	struct stat s;
+	GStatBuf s;
 
 	if (g_stat(file, &s) < 0) {
 		FILE_OP_ERROR(file, "stat");
@@ -2140,7 +2140,7 @@
 
 time_t get_file_mtime(const gchar *file)
 {
-	struct stat s;
+	GStatBuf s;
 
 	if (g_stat(file, &s) < 0) {
 		FILE_OP_ERROR(file, "stat");
@@ -2178,7 +2178,7 @@
 
 gboolean file_exist(const gchar *file, gboolean allow_fifo)
 {
-	struct stat s;
+	GStatBuf s;
 
 	if (file == NULL)
 		return FALSE;
@@ -2486,7 +2486,7 @@
 
 gint remove_dir_recursive(const gchar *dir)
 {
-	struct stat s;
+	GStatBuf s;
 	GDir *dp;
 	const gchar *dir_name;
 	gchar *prev_dir;
@@ -3235,7 +3235,7 @@
 {
 	FILE *fp;
 	gchar *str;
-	struct stat s;
+	GStatBuf s;
 #ifndef G_OS_WIN32
 	gint fd, err;
 	struct timeval timeout = {1, 0};
@@ -5328,7 +5328,7 @@
 
 int claws_unlink(const gchar *filename) 
 {
-	struct stat s;
+	GStatBuf s;
 	static int found_shred = -1;
 	static const gchar *args[4];
 
@@ -5419,7 +5419,7 @@
 {
 	gchar **path_parts;
 	GSList *canonical_parts = NULL;
-	struct stat st;
+	GStatBuf st;
 	int i;
 	gboolean follow_symlinks = TRUE;
 
diff -ur claws-mail-3.11.1-orig/src/common/w32lib.h claws-mail-3.11.1/src/common/w32lib.h
--- claws-mail-3.11.1-orig/src/common/w32lib.h	2015-06-06 18:01:32.958681041 +0200
+++ claws-mail-3.11.1/src/common/w32lib.h	2015-06-06 18:22:21.692219685 +0200
@@ -84,6 +84,8 @@
 #include <sys/time.h>
 #endif
 
+#include <glib/gstdio.h>
+
 /* Mingw32 3.4.4 defines interface to struct and thus breaks our own
    use of that symbol.  Undef it here. */
 #if defined(_BASETYPS_H) && defined(interface) 
@@ -174,7 +176,7 @@
 #endif
 
 /*** stat ***/
-int lstat( const char *file_name, struct stat *buf );
+int lstat( const char *file_name, GStatBuf *buf );
 
 /*** sys/wait ***/
 pid_t waitpid( pid_t pid, int *status, int options );
diff -ur claws-mail-3.11.1-orig/src/common/w32_stat.c claws-mail-3.11.1/src/common/w32_stat.c
--- claws-mail-3.11.1-orig/src/common/w32_stat.c	2015-06-06 18:01:32.958681041 +0200
+++ claws-mail-3.11.1/src/common/w32_stat.c	2015-06-06 18:03:28.765796464 +0200
@@ -24,7 +24,7 @@
 
 #include "w32lib.h"
 
-int lstat( const char *file_name, struct stat *buf ){
-  return stat( file_name, buf );
+int lstat( const char *file_name, GStatBuf *buf ){
+  return g_stat( file_name, buf );
 }
 
diff -ur claws-mail-3.11.1-orig/src/compose.c claws-mail-3.11.1/src/compose.c
--- claws-mail-3.11.1-orig/src/compose.c	2015-06-06 18:01:32.962681149 +0200
+++ claws-mail-3.11.1/src/compose.c	2015-06-06 18:05:41.853371482 +0200
@@ -3575,7 +3575,7 @@
 	gint len;
 	FILE *fp;
 	gboolean prev_autowrap;
-	struct stat file_stat;
+	GStatBuf file_stat;
 	int ret;
 	GString *file_contents = NULL;
 
@@ -6114,7 +6114,7 @@
 	AttachInfo *ainfo;
 	GtkTreeView *tree_view = GTK_TREE_VIEW(compose->attach_clist);
 	MimeInfo *mimepart;
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gchar *type, *subtype;
 	GtkTreeModel *model;
 	GtkTreeIter iter;
@@ -9974,7 +9974,7 @@
 		compose_close(compose);
 		return TRUE;
 	} else {
-		struct stat s;
+		GStatBuf s;
 		gchar *path;
 
 		path = folder_item_fetch_msg(draft, msgnum);
diff -ur claws-mail-3.11.1-orig/src/jpilot.c claws-mail-3.11.1/src/jpilot.c
--- claws-mail-3.11.1-orig/src/jpilot.c	2015-06-06 18:01:32.970681366 +0200
+++ claws-mail-3.11.1/src/jpilot.c	2015-06-06 18:07:09.115715361 +0200
@@ -353,7 +353,7 @@
  */
 static gboolean jpilot_mark_files( JPilotFile *pilotFile ) {
 	gboolean retVal = FALSE;
-	struct stat filestat;
+	GStatBuf filestat;
 	gchar *pcFile;
 
 	/* Mark PDB file cache */
@@ -381,7 +381,7 @@
  */
 static gboolean jpilot_check_files( JPilotFile *pilotFile ) {
 	gboolean retVal = TRUE;
-	struct stat filestat;
+	GStatBuf filestat;
 	gchar *pcFile;
 
 	/* Check main file */
diff -ur claws-mail-3.11.1-orig/src/mh.c claws-mail-3.11.1/src/mh.c
--- claws-mail-3.11.1-orig/src/mh.c	2015-06-06 18:01:32.974681474 +0200
+++ claws-mail-3.11.1/src/mh.c	2015-06-06 18:07:53.540908573 +0200
@@ -212,7 +212,7 @@
 gboolean mh_scan_required(Folder *folder, FolderItem *item)
 {
 	gchar *path;
-	struct stat s;
+	GStatBuf s;
 
 	path = folder_item_get_path(item);
 	cm_return_val_if_fail(path != NULL, FALSE);
@@ -727,7 +727,7 @@
 static gboolean mh_is_msg_changed(Folder *folder, FolderItem *item,
 				  MsgInfo *msginfo)
 {
-	struct stat s;
+	GStatBuf s;
 
 	if (g_stat(itos(msginfo->msgnum), &s) < 0 ||
 	    msginfo->size  != s.st_size || (
@@ -1103,7 +1103,7 @@
 	struct dirent *d;
 #endif
 	const gchar *dir_name;
-	struct stat s;
+	GStatBuf s;
  	gchar *real_path, *entry, *utf8entry, *utf8name;
 	gint n_msg = 0;
 
@@ -1427,7 +1427,7 @@
 
 static void mh_set_mtime(Folder *folder, FolderItem *item)
 {
-	struct stat s;
+	GStatBuf s;
 	gchar *path = folder_item_get_path(item);
 
 	cm_return_if_fail(path != NULL);
diff -ur claws-mail-3.11.1-orig/src/plugins/archive/archiver_gtk.c claws-mail-3.11.1/src/plugins/archive/archiver_gtk.c
--- claws-mail-3.11.1-orig/src/plugins/archive/archiver_gtk.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/archive/archiver_gtk.c	2015-06-06 18:09:42.887845373 +0200
@@ -659,7 +659,7 @@
 		N_COLUMNS
 	};
 
-	struct stat st;
+	GStatBuf st;
 	GtkListStore* list;
 	GtkTreeIter iter;
 	GtkTreeView* view;
diff -ur claws-mail-3.11.1-orig/src/plugins/archive/libarchive_archive.c claws-mail-3.11.1/src/plugins/archive/libarchive_archive.c
--- claws-mail-3.11.1-orig/src/plugins/archive/libarchive_archive.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/archive/libarchive_archive.c	2015-06-06 18:09:29.223478390 +0200
@@ -462,7 +462,7 @@
 	char* buf = NULL;
 	ssize_t len;
 	int fd;
-	struct stat st;
+	GStatBuf st;
 	struct file_info* file;
 	gchar* filename = NULL;
 	gchar* msg = NULL;
@@ -598,7 +598,7 @@
 
 #ifdef _TEST
 void archive_scan_folder(const char* dir) {
-	struct stat st;
+	GStatBuf st;
 	DIR* root;
 	struct dirent* ent;
 	gchar cwd[PATH_MAX];
diff -ur claws-mail-3.11.1-orig/src/plugins/clamd/clamav_plugin.c claws-mail-3.11.1/src/plugins/clamd/clamav_plugin.c
--- claws-mail-3.11.1-orig/src/plugins/clamd/clamav_plugin.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/clamd/clamav_plugin.c	2015-06-06 18:08:54.530546636 +0200
@@ -85,7 +85,7 @@
 	gchar *outfile;
 	response buf;
 	int max;
-	struct stat info;
+	GStatBuf info;
 	gchar* msg;
 
 	outfile = procmime_get_tmp_file_name(mimeinfo);
@@ -93,7 +93,7 @@
 		g_warning("Can't get the part of multipart message.");
 	else {
     	max = config.clamav_max_size * 1048576; /* maximum file size */
-		if (stat(outfile, &info) == -1)
+		if (g_stat(outfile, &info) == -1)
 			g_warning("Can't determine file size");
 		else {
 			if (info.st_size <= max) {
diff -ur claws-mail-3.11.1-orig/src/plugins/clamd/clamav_plugin_gtk.c claws-mail-3.11.1/src/plugins/clamd/clamav_plugin_gtk.c
--- claws-mail-3.11.1-orig/src/plugins/clamd/clamav_plugin_gtk.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/clamd/clamav_plugin_gtk.c	2015-06-06 18:08:28.813855943 +0200
@@ -104,7 +104,7 @@
 }
 
 static void check_permission(gchar* folder) {
-	struct stat info;
+	GStatBuf info;
 
 	if (g_stat(folder, &info) < 0)
 		return;
diff -ur claws-mail-3.11.1-orig/src/plugins/geolocation/geolocation_plugin.c claws-mail-3.11.1/src/plugins/geolocation/geolocation_plugin.c
--- claws-mail-3.11.1-orig/src/plugins/geolocation/geolocation_plugin.c	2015-06-06 18:01:32.926680180 +0200
+++ claws-mail-3.11.1/src/plugins/geolocation/geolocation_plugin.c	2015-06-06 18:11:09.258164925 +0200
@@ -150,7 +150,7 @@
 static gchar* get_ip_from_msginfo(MsgInfo *msginfo)
 {
   gchar *file;
-  struct stat ss;
+  GStatBuf ss;
   FILE *fp;
   gchar buf[BUFFSIZE];
   gint hnum;
diff -ur claws-mail-3.11.1-orig/src/plugins/libravatar/libravatar.c claws-mail-3.11.1/src/plugins/libravatar/libravatar.c
--- claws-mail-3.11.1-orig/src/plugins/libravatar/libravatar.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/libravatar/libravatar.c	2015-06-06 18:11:25.014588064 +0200
@@ -210,7 +210,7 @@
 
 static gboolean is_recent_enough(const gchar *filename)
 {
-	struct stat s;
+	GStatBuf s;
 	time_t t;
 
 	if (libravatarprefs.cache_icons) {
diff -ur claws-mail-3.11.1-orig/src/plugins/mailmbox/maillock.c claws-mail-3.11.1/src/plugins/mailmbox/maillock.c
--- claws-mail-3.11.1-orig/src/plugins/mailmbox/maillock.c	2015-06-06 18:01:32.930680287 +0200
+++ claws-mail-3.11.1/src/plugins/mailmbox/maillock.c	2015-06-06 18:10:58.089864998 +0200
@@ -171,7 +171,7 @@
   time(&start);
   while (1) {
     int fd;
-    struct stat st;
+    GStatBuf st;
     time_t now;
     
     /* global timeout */
@@ -193,7 +193,7 @@
     /* libEtPan! - adds a delay of 5 seconds between each tries */
     sleep(5);
     
-    if (stat(lockfilename, &st) < 0) {
+    if (g_stat(lockfilename, &st) < 0) {
       if (statfailed++ > 5) {
 	res = -1;
 	goto unlock;
diff -ur claws-mail-3.11.1-orig/src/plugins/mailmbox/mailmbox.c claws-mail-3.11.1/src/plugins/mailmbox/mailmbox.c
--- claws-mail-3.11.1-orig/src/plugins/mailmbox/mailmbox.c	2015-06-06 18:01:32.930680287 +0200
+++ claws-mail-3.11.1/src/plugins/mailmbox/mailmbox.c	2015-06-06 18:10:23.728942212 +0200
@@ -130,11 +130,11 @@
 int claws_mailmbox_map(struct claws_mailmbox_folder * folder)
 {
   char * str;
-  struct stat buf;
+  GStatBuf buf;
   int res;
   int r;
 
-  r = stat(folder->mb_filename, &buf);
+  r = g_stat(folder->mb_filename, &buf);
   if (r < 0) {
     debug_print("stat failed %d\n", r);
     res = MAILMBOX_ERROR_FILE;
@@ -187,9 +187,9 @@
 void claws_mailmbox_timestamp(struct claws_mailmbox_folder * folder)
 {
   int r;
-  struct stat buf;
+  GStatBuf buf;
 
-  r = stat(folder->mb_filename, &buf);
+  r = g_stat(folder->mb_filename, &buf);
   if (r < 0)
     folder->mb_mtime = (time_t) -1;
   else
@@ -238,11 +238,11 @@
     int (* custom_lock)(struct claws_mailmbox_folder *),
     int (* custom_unlock)(struct claws_mailmbox_folder *))
 {
-  struct stat buf;
+  GStatBuf buf;
   int res;
   int r;
 
-  r = stat(folder->mb_filename, &buf);
+  r = g_stat(folder->mb_filename, &buf);
   if (r < 0) {
     buf.st_mtime = (time_t) -1;
   }
diff -ur claws-mail-3.11.1-orig/src/plugins/perl/perl_plugin.c claws-mail-3.11.1/src/plugins/perl/perl_plugin.c
--- claws-mail-3.11.1-orig/src/plugins/perl/perl_plugin.c	2015-06-06 18:01:32.938680503 +0200
+++ claws-mail-3.11.1/src/plugins/perl/perl_plugin.c	2015-06-06 18:12:18.276018376 +0200
@@ -248,13 +248,13 @@
 {
   gboolean retVal;
   gchar *indexfile;
-  struct stat filestat;
+  GStatBuf filestat;
 
   if(tl->g_slist == NULL)
     return TRUE;
 
   indexfile = g_strconcat(get_rc_dir(), G_DIR_SEPARATOR_S, ADDRESSBOOK_INDEX_FILE, NULL);
-  if((stat(indexfile,&filestat) == 0) && filestat.st_mtime <= tl->mtime)
+  if((g_stat(indexfile,&filestat) == 0) && filestat.st_mtime <= tl->mtime)
      retVal = FALSE;
   else
     retVal = TRUE;
@@ -267,7 +267,7 @@
 static void init_email_slist(void)
 {
   gchar *indexfile;
-  struct stat filestat;
+  GStatBuf filestat;
 
   if(email_slist->g_slist != NULL) {
     free_PerlPluginEmailEntry_slist(email_slist->g_slist);
@@ -277,7 +277,7 @@
   addrindex_load_person_attribute(NULL,add_to_email_slist);
 
   indexfile = g_strconcat(get_rc_dir(), G_DIR_SEPARATOR_S, ADDRESSBOOK_INDEX_FILE, NULL);
-  if(stat(indexfile,&filestat) == 0)
+  if(g_stat(indexfile,&filestat) == 0)
     email_slist->mtime = filestat.st_mtime;
   g_free(indexfile);
   debug_print("Initialisation of email slist completed\n");
@@ -424,7 +424,7 @@
 {
   PerlPluginTimedSList *tl;
   gchar *indexfile;
-  struct stat filestat;
+  GStatBuf filestat;
 
   /* Check if key exists. Free it if it does. */
   if((tl = g_hash_table_lookup(attribute_hash,attr)) != NULL) {
@@ -445,7 +445,7 @@
   addrindex_load_person_attribute(attribute_key,add_to_attribute_hash);
 
   indexfile = g_strconcat(get_rc_dir(), G_DIR_SEPARATOR_S, ADDRESSBOOK_INDEX_FILE, NULL);
-  if(stat(indexfile,&filestat) == 0)
+  if(g_stat(indexfile,&filestat) == 0)
     tl->mtime = filestat.st_mtime;
   g_free(indexfile);
 
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/parse822.c claws-mail-3.11.1/src/plugins/rssyl/parse822.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/parse822.c	2015-06-06 18:01:32.954680934 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/parse822.c	2015-06-06 18:12:43.976708537 +0200
@@ -243,7 +243,7 @@
 	gchar *path = NULL, *fname = NULL;
 	DIR *dp;
 	struct dirent *d;
-	struct stat st;
+	GStatBuf st;
 	gint num;
 	FeedItem *item = NULL;
 	RFeedCtx *ctx;
diff -ur claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl.c claws-mail-3.11.1/src/plugins/rssyl/rssyl.c
--- claws-mail-3.11.1-orig/src/plugins/rssyl/rssyl.c	2015-06-06 18:01:32.954680934 +0200
+++ claws-mail-3.11.1/src/plugins/rssyl/rssyl.c	2015-06-06 18:12:28.396290144 +0200
@@ -650,7 +650,7 @@
 static gboolean rssyl_is_msg_changed(Folder *folder, FolderItem *item,
 		MsgInfo *msginfo)
 {
-	struct stat s;
+	GStatBuf s;
 	gchar *path = NULL;
 
 	g_return_val_if_fail(folder != NULL, FALSE);
diff -ur claws-mail-3.11.1-orig/src/plugins/tnef_parse/tnef_parse.c claws-mail-3.11.1/src/plugins/tnef_parse/tnef_parse.c
--- claws-mail-3.11.1-orig/src/plugins/tnef_parse/tnef_parse.c	2015-06-06 18:01:32.938680503 +0200
+++ claws-mail-3.11.1/src/plugins/tnef_parse/tnef_parse.c	2015-06-06 18:13:33.406035874 +0200
@@ -50,7 +50,7 @@
 	MimeInfo *sub_info = NULL;
 	gchar *tmpfilename = NULL;
 	FILE *fp = get_tmpfile_in_dir(get_mime_tmp_dir(), &tmpfilename);
-	struct stat statbuf;
+	GStatBuf statbuf;
 
 	if (!fp) {
 		g_free(tmpfilename);
@@ -87,7 +87,7 @@
 	MimeInfo *sub_info = NULL;
 	gchar *tmpfilename = NULL;
 	FILE *fp = get_tmpfile_in_dir(get_mime_tmp_dir(), &tmpfilename);
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gchar *content_type = NULL;
 	if (!fp) {
 		g_free(tmpfilename);
@@ -141,7 +141,7 @@
 	MimeInfo *sub_info = NULL;
 	gchar *tmpfilename = NULL;
 	FILE *fp = get_tmpfile_in_dir(get_mime_tmp_dir(), &tmpfilename);
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gboolean result = FALSE;
 	if (!fp) {
 		g_free(tmpfilename);
@@ -181,7 +181,7 @@
 	MimeInfo *sub_info = NULL;
 	gchar *tmpfilename = NULL;
 	FILE *fp = get_tmpfile_in_dir(get_mime_tmp_dir(), &tmpfilename);
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gboolean result = FALSE;
 	if (!fp) {
 		g_free(tmpfilename);
@@ -234,7 +234,7 @@
 	MimeInfo *sub_info = NULL;
 	gchar *tmpfilename = NULL;
 	FILE *fp = get_tmpfile_in_dir(get_mime_tmp_dir(), &tmpfilename);
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gboolean result = FALSE;
 	if (!fp) {
 		g_free(tmpfilename);
diff -ur claws-mail-3.11.1-orig/src/plugins/vcalendar/vcal_folder.c claws-mail-3.11.1/src/plugins/vcalendar/vcal_folder.c
--- claws-mail-3.11.1-orig/src/plugins/vcalendar/vcal_folder.c	2015-06-06 18:01:32.942680612 +0200
+++ claws-mail-3.11.1/src/plugins/vcalendar/vcal_folder.c	2015-06-06 18:14:06.430922677 +0200
@@ -1124,7 +1124,7 @@
 
 static gboolean vcal_scan_required(Folder *folder, FolderItem *item)
 {
-	struct stat s;
+	GStatBuf s;
 	VCalFolderItem *vitem = (VCalFolderItem *)item;
 
 	g_return_val_if_fail(item != NULL, FALSE);
@@ -1144,7 +1144,7 @@
 
 static void vcal_set_mtime(Folder *folder, FolderItem *item)
 {
-	struct stat s;
+	GStatBuf s;
 	gchar *path = folder_item_get_path(item);
 
 	if (folder->inbox != item)
diff -ur claws-mail-3.11.1-orig/src/prefs_themes.c claws-mail-3.11.1/src/prefs_themes.c
--- claws-mail-3.11.1-orig/src/prefs_themes.c	2015-06-06 18:01:32.966681258 +0200
+++ claws-mail-3.11.1/src/prefs_themes.c	2015-06-06 18:14:23.791388842 +0200
@@ -159,7 +159,7 @@
 
 static void prefs_themes_file_stats(const gchar *filename, gpointer data)
 {
-	struct stat s;
+	GStatBuf s;
 	DirInfo    *di = (DirInfo *)data;
 	gint        len;
 	
diff -ur claws-mail-3.11.1-orig/src/procheader.c claws-mail-3.11.1/src/procheader.c
--- claws-mail-3.11.1-orig/src/procheader.c	2015-06-06 18:01:32.970681366 +0200
+++ claws-mail-3.11.1/src/procheader.c	2015-06-06 18:14:32.287616984 +0200
@@ -337,7 +337,7 @@
 MsgInfo *procheader_parse_file(const gchar *file, MsgFlags flags,
 			       gboolean full, gboolean decrypted)
 {
-	struct stat s;
+	GStatBuf s;
 	FILE *fp;
 	MsgInfo *msginfo;
 
diff -ur claws-mail-3.11.1-orig/src/procmime.c claws-mail-3.11.1/src/procmime.c
--- claws-mail-3.11.1-orig/src/procmime.c	2015-06-06 18:01:32.966681258 +0200
+++ claws-mail-3.11.1/src/procmime.c	2015-06-06 18:15:08.048577241 +0200
@@ -315,7 +315,7 @@
 	gint readend;
 	gchar *tmpfilename;
 	FILE *outfp, *infp;
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gboolean tmp_file = FALSE;
 	gboolean flowed = FALSE;
 	gboolean delsp = FALSE; 
@@ -529,7 +529,7 @@
 	FILE *infp = NULL, *outfp;
 	gint len;
 	gchar *tmpfilename;
-	struct stat statbuf;
+	GStatBuf statbuf;
 	gboolean err = FALSE;
 
 	if (mimeinfo->content == MIMECONTENT_EMPTY)
@@ -2165,7 +2165,7 @@
 static MimeInfo *procmime_scan_file_with_offset(const gchar *filename, int offset, gboolean short_scan)
 {
 	MimeInfo *mimeinfo;
-	struct stat buf;
+	GStatBuf buf;
 
 	if (g_stat(filename, &buf) < 0) {
 		FILE_OP_ERROR(filename, "stat");
