#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/plugins/libravatar/libravatar.c claws-mail-3.11.1/src/plugins/libravatar/libravatar.c
--- claws-mail-3.11.1-orig/src/plugins/libravatar/libravatar.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/libravatar/libravatar.c	2015-06-08 18:22:57.033825458 +0200
@@ -286,7 +286,7 @@
 			return FALSE;
 		}
 		/* not cached copy: try network */
-		if (prefs_common.work_offline) {
+		if (prefs_common_get_prefs()->work_offline) {
 			debug_print("working off-line: libravatar network retrieval skipped\n");
 			return FALSE;
 		}
diff -ur claws-mail-3.11.1-orig/src/plugins/libravatar/libravatar_prefs.c claws-mail-3.11.1/src/plugins/libravatar/libravatar_prefs.c
--- claws-mail-3.11.1-orig/src/plugins/libravatar/libravatar_prefs.c	2015-06-06 18:01:32.934680394 +0200
+++ claws-mail-3.11.1/src/plugins/libravatar/libravatar_prefs.c	2015-06-08 18:22:35.945258422 +0200
@@ -178,10 +178,10 @@
 			TRUE);
 
 	if (mode == DEF_MODE_NONE) {
-		prefs_common.enable_avatars = AVATARS_ENABLE_BOTH;
+		prefs_common_get_prefs()->enable_avatars = AVATARS_ENABLE_BOTH;
 	} else {
 		/* don't waste time with headers that won't be displayed */
-		prefs_common.enable_avatars = AVATARS_DISABLE;
+		prefs_common_get_prefs()->enable_avatars = AVATARS_DISABLE;
 		/* empty missing cache when switching to generated */
 		g_hash_table_remove_all(libravatarmisses);
 	}
@@ -242,8 +242,9 @@
 		libravatarprefs.default_mode = DEF_MODE_NONE;
 	}
 	/* don't waste time with headers that won't be displayed */
-	prefs_common.enable_avatars = (libravatarprefs.default_mode == DEF_MODE_NONE)
-						? AVATARS_ENABLE_BOTH: AVATARS_DISABLE;
+	prefs_common_get_prefs()->enable_avatars =
+		(libravatarprefs.default_mode == DEF_MODE_NONE)
+		? AVATARS_ENABLE_BOTH: AVATARS_DISABLE;
 
 	label = gtk_label_new(_("URL:"));
 	gtk_widget_show(label);
