#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/plugins/fancy/claws.def claws-mail-3.11.1/src/plugins/fancy/claws.def
--- claws-mail-3.11.1-orig/src/plugins/fancy/claws.def	2015-06-06 18:01:32.950680827 +0200
+++ claws-mail-3.11.1/src/plugins/fancy/claws.def	2015-06-14 11:31:56.289497243 +0200
@@ -18,6 +18,8 @@
 end_address_completion
 execute_command_line
 extract_address
+file_exist
+filesel_select_file_open
 filesel_select_file_save
 folder_find_item_from_identifier
 folder_get_default_trash
@@ -30,6 +32,7 @@
 get_locale_dir
 get_rc_dir
 gtkut_get_browse_directory_btn
+gtkut_get_browse_file_btn
 gtkut_get_options_frame
 gtkutils_scroll_one_line
 gtkutils_scroll_page
@@ -47,6 +50,7 @@
 mimeview_register_viewer_factory
 mimeview_unregister_viewer_factory
 noticeview_hide
+open_txt_editor
 open_uri
 plugin_get_loaded_by_name
 pref_get_escaped_pref
@@ -55,6 +59,7 @@
 prefs_button_toggled
 prefs_button_toggled follow
 prefs_common
+prefs_common_get_ext_editor_cmd
 prefs_common_get_prefs
 prefs_common_get_uri_cmd
 pref_set_entry_from_pref
@@ -85,3 +90,4 @@
 statusbar_pop_all
 statusbar_print_all
 statusbar_progress_all
+str_write_to_file
