#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/common/w32_dirent.c claws-mail-3.11.1/src/common/w32_dirent.c
--- claws-mail-3.11.1-orig/src/common/w32_dirent.c	2015-06-18 15:10:11.363527840 +0200
+++ claws-mail-3.11.1/src/common/w32_dirent.c	2015-06-18 15:11:17.361306339 +0200
@@ -25,76 +25,3 @@
 #include <dirent.h>
 
 #include "w32lib.h"
-
-
-
-DIR *
-opendir ( const char *name )
-{
-  DIR *dir;
-  
-  dir = calloc (1, sizeof *dir + strlen (name));
-  if (!dir)
-    return NULL;
-  strcpy (dir->dd_name, name);
-  return dir;
-}
-
-int 
-closedir (DIR *dir)
-{
-  FindClose( (HANDLE)dir->dd_handle );
-  free( dir );
-  return 0;
-}
-
-struct dirent *
-readdir( DIR *dir )
-{
-  WIN32_FIND_DATA fInfo;
-  struct dirent *xdirent;
-  int ret;
-  
-  if ( !dir->dd_handle )
-    {
-      char *dirname;
-      
-      if (*dir->dd_name) 
-        {
-          size_t n = strlen (dir->dd_name);
-          dirname = malloc (n + 4 + 1);
-          if (dirname) {
-            strcpy (dirname, dir->dd_name);
-            strcpy (dirname + n, "\\*.*");
-          }
-        }
-      else
-        dirname = strdup( "\\*.*" );
-      if (!dirname)
-        return NULL; /* Error. */
-
-      dir->dd_handle = (long)FindFirstFile( dirname, &fInfo );
-      free( dirname );
-      if ( !dir->dd_handle )
-        ret = 0;
-      else
-        ret = 1;
-    } 
-  else if ( dir->dd_handle != -1l )
-    {
-        ret = FindNextFile ((HANDLE)dir->dd_handle, &fInfo);
-    }
-  else
-    ret = 0;
-  if ( !ret ) 
-    return NULL;
-
-  xdirent = calloc ( 1, sizeof *xdirent);
-  if (xdirent)
-    {
-      strncpy (xdirent->d_name, fInfo.cFileName, FILENAME_MAX -1 );
-      xdirent->d_name[FILENAME_MAX-1] = 0;
-      xdirent->d_namlen = strlen( xdirent->d_name );
-    }
-  return xdirent;
-}
diff -ur claws-mail-3.11.1-orig/src/common/w32lib.h claws-mail-3.11.1/src/common/w32lib.h
--- claws-mail-3.11.1-orig/src/common/w32lib.h	2015-06-18 15:10:11.363527840 +0200
+++ claws-mail-3.11.1/src/common/w32lib.h	2015-06-18 15:10:48.728534778 +0200
@@ -161,11 +161,6 @@
 
 #endif /* !__MINGW32__ */
 
-
-DIR *opendir( const char *name );
-int closedir( DIR *dir );
-struct dirent *readdir( DIR *dir );
-
 #if defined (__MINGW32__) && MINGW32_VERSION < 312
 struct timezone {
   int tz_minuteswest;
