#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ur claws-mail-3.11.1-orig/src/ldaputil.c claws-mail-3.11.1/src/ldaputil.c
--- claws-mail-3.11.1-orig/src/ldaputil.c	2015-06-18 15:10:11.383528379 +0200
+++ claws-mail-3.11.1/src/ldaputil.c	2015-06-20 22:32:02.089469665 +0200
@@ -324,7 +324,14 @@
 		strncpy2(error, ld_error, sizeof(error));
 	else
 		strncpy2(error, _("Unknown error"), sizeof(error));
+#ifndef G_OS_WIN32
+	/* From https://msdn.microsoft.com/en-us/library/aa366594%28v=vs.85%29.aspx
+	 * "LDAP_OPT_ERROR_STRING returns a pointer to an internal static
+	 * string table, and ldap_memfree should not be called when using
+	 * this session option."
+	 */
 	ldap_memfree(ld_error);
+#endif
 
 	return error;
 }
