#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -Naur a/WebKit/gtk/webkit/webkitwebframe.cpp webkitgtk-1.10.2c/Source/WebKit/gtk/webkit/webkitwebframe.cpp
--- a/WebKit/gtk/webkit/webkitwebframe.cpp	2012-11-23 15:12:18 -0500
+++ b/WebKit/gtk/webkit/webkitwebframe.cpp	2013-07-08 10:15:06 -0400
@@ -912,8 +912,13 @@
 {
     PrintContext* printContext = reinterpret_cast<PrintContext*>(user_data);
 
+    gdouble x_res = gtk_print_context_get_dpi_x(context);
+    gdouble y_res = gtk_print_context_get_dpi_y(context);
     float width = gtk_print_context_get_width(context);
     float height = gtk_print_context_get_height(context);
+    // force 72 dpi scale for all platforms
+    width /= x_res/72.0;
+    height /= y_res/72.0;
     FloatRect printRect = FloatRect(0, 0, width, height);
 
     printContext->begin(width);
