#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN claws-mail-3.12.0-orig/src/plugins/managesieve/claws.def claws-mail-3.12.0/src/plugins/managesieve/claws.def
--- claws-mail-3.12.0-orig/src/plugins/managesieve/claws.def	2015-07-19 22:29:03.832080060 +0200
+++ claws-mail-3.12.0/src/plugins/managesieve/claws.def	2015-07-19 22:31:45.760373393 +0200
@@ -12,6 +12,7 @@
 combobox_select_by_data
 debug_print_real
 debug_srcname
+get_rc_dir
 gtkut_get_options_frame
 gtkut_sc_combobox_create
 gtkut_stock_button_set_create
@@ -40,6 +41,13 @@
 prefs_account_unregister_page
 prefs_button_toggled
 prefs_common_get_prefs
+prefs_file_close
+prefs_file_close_revert
+prefs_read_config
+prefs_set_block_label
+prefs_set_default
+prefs_write_open
+prefs_write_param
 session_connect
 session_destroy
 session_disconnect
