#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN claws-mail-3.17.1-orig/src/main.c claws-mail-3.17.1/src/main.c
--- claws-mail-3.17.1-orig/src/main.c	2018-08-26 13:25:23.000000000 +0200
+++ claws-mail-3.17.1/src/main.c	2018-08-30 19:45:26.588272531 +0200
@@ -136,7 +136,9 @@
 #ifdef HAVE_LIBETPAN
 #include "imap-thread.h"
 #include "nntp-thread.h"
+extern void mailsasl_set_plugin_path(char *path);
 #endif
+
 #include "stock_pixmap.h"
 #ifdef USE_GNUTLS
 #  include "ssl.h"
@@ -1343,6 +1345,13 @@
 	}
 
 #ifdef HAVE_LIBETPAN
+	/* Following block utilizes unofficial libetpan API, added
+	 * downstream in our claws-win32-installer repository. */
+	gchar *pluginpath = g_strdup_printf("%s\\lib\\sasl2",
+			claws_get_startup_dir());
+	mailsasl_set_plugin_path(pluginpath);
+	g_free(pluginpath);
+
 	imap_main_init(prefs_common.skip_ssl_cert_check);
 	imap_main_set_timeout(prefs_common.io_timeout_secs);
 	nntp_main_init(prefs_common.skip_ssl_cert_check);
