#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN claws-mail-3.17.0-orig/manual/dist/pdf/Makefile.in claws-mail-3.17.0/manual/dist/pdf/Makefile.in
--- claws-mail-3.17.0-orig/manual/dist/pdf/Makefile.in	2018-08-15 19:45:00.000000000 +0200
+++ claws-mail-3.17.0/manual/dist/pdf/Makefile.in	2018-08-23 13:38:38.095917109 +0200
@@ -647,7 +647,7 @@
 @MANUAL_PDF_TRUE@	../../plugins.xml \
 @MANUAL_PDF_TRUE@	../../starting.xml \
 @MANUAL_PDF_TRUE@	../../claws-mail-manual.xml
-@MANUAL_PDF_TRUE@	-docbook2pdf -o . ../../claws-mail-manual.xml
+@MANUAL_PDF_TRUE@	-docbook2pdf -o . "$(srcdir)/../../claws-mail-manual.xml"
 
 @MANUAL_PDF_TRUE@maintainer-clean:
 @MANUAL_PDF_TRUE@	-rm claws-mail-manual.pdf claws-mail-manual.fo claws-mail-manual.xsl
