#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN claws-mail-3.16.0-orig/src/compose.c claws-mail-3.16.0/src/compose.c
--- claws-mail-3.16.0-orig/src/compose.c	2017-12-17 12:48:38.000000000 +0100
+++ claws-mail-3.16.0/src/compose.c	2018-02-28 17:36:17.839262350 +0100
@@ -10526,7 +10526,7 @@
 		GFile *f;
 		GFileInfo *fi;
 		GTimeVal tv;
-		GError *error;
+		GError *error = NULL;
 #else
 		GStatBuf s;
 #endif
@@ -10555,7 +10555,6 @@
 		mtime = tv.tv_sec;
 		g_object_unref(fi);
 		g_object_unref(f);
-		g_free(path);
 #else
 		if (g_stat(path, &s) < 0) {
 			FILE_OP_ERROR(path, "stat");
