#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff --git a/src/plugins/litehtml_viewer/claws.def b/src/plugins/litehtml_viewer/claws.def
index 90b9af383..30dada9e5 100644
--- a/src/plugins/litehtml_viewer/claws.def
+++ b/src/plugins/litehtml_viewer/claws.def
@@ -5,6 +5,8 @@ debug_srcname
 get_rc_dir
 gtkut_get_options_frame
 gtkut_widget_draw_now
+gtkutils_scroll_one_line
+gtkutils_scroll_page
 mainwindow_get_mainwindow
 mimeview_register_viewer_factory
 mimeview_unregister_viewer_factory
