#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN glib-networking-2.48.2-orig/configure glib-networking-2.48.2/configure
--- glib-networking-2.48.2-orig/configure	2016-05-09 16:03:18.000000000 +0200
+++ glib-networking-2.48.2/configure	2016-07-09 11:57:00.571747092 +0200
@@ -14626,7 +14626,7 @@
 
 GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_46"
 
-GIO_MODULE_DIR=$($PKG_CONFIG --variable giomoduledir gio-2.0)
+GIO_MODULE_DIR=$($PKG_CONFIG --define-variable=prefix=${prefix} --variable giomoduledir gio-2.0)
 if test "$GIO_MODULE_DIR" = ""; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
