#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN gnutls-3.5.2-orig/src/socket.c gnutls-3.5.2/src/socket.c
--- gnutls-3.5.2-orig/src/socket.c	2016-07-08 23:18:19.887794943 +0200
+++ gnutls-3.5.2/src/socket.c	2016-07-08 23:18:45.667205089 +0200
@@ -281,7 +281,6 @@
 	if (s != NULL) {
 		return s->s_port;
 	}
-	endservent();
 
 	return 443;
 }
@@ -296,7 +295,6 @@
 	if (s != NULL) {
 		return s->s_name;
 	}
-	endservent();
 
 	return "443";
 }
