#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN libsoup-2.40.3-orig/Makefile.in libsoup-2.40.3/Makefile.in
--- libsoup-2.40.3-orig/Makefile.in	2017-01-02 11:45:14.610480816 +0100
+++ libsoup-2.40.3/Makefile.in	2017-01-02 11:45:49.745792910 +0100
@@ -370,7 +370,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
-SUBDIRS = libsoup po tests examples docs win32
+SUBDIRS = libsoup po win32
 EXTRA_DIST = \
 	data/effective_tld_names.dat	\
 	libsoup-2.4.pc.in		\
