#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN gtk+-2.24.32-orig/Makefile.in gtk+-2.24.32/Makefile.in
--- gtk+-2.24.32-orig/Makefile.in	2018-01-08 22:35:47.000000000 +0100
+++ gtk+-2.24.32/Makefile.in	2018-08-29 17:48:09.923024027 +0200
@@ -568,7 +568,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+SRC_SUBDIRS = gdk gtk modules perf
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
