#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN libetpan-1.7-orig/configure libetpan-1.7/configure
--- libetpan-1.7-orig/configure	2016-11-24 21:47:29.000000000 +0100
+++ libetpan-1.7/configure	2016-11-24 21:49:11.657414504 +0100
@@ -18018,10 +18018,11 @@
 #define INET6
             #include <sys/types.h>
             #include <netinet/in.h>
+						#include <ws2tcpip.h>
 int
 main ()
 {
-int x = IPPROTO_IPV6; struct in6_addr a;
+struct in6_addr a;
   ;
   return 0;
 }
@@ -18044,35 +18045,6 @@
 $as_echo "#define HAVE_IPV6 1" >>confdefs.h
 
 
-        for func in "getaddrinfo" "freeaddrinfo"; do
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $func" >&5
-$as_echo_n "checking for $func... " >&6; }
-      cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <sys/socket.h>
-                   #include <netdb.h>
-int
-main ()
-{
-(void)$func;
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  func_present=yes
-else
-  func_present=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: $func_present" >&5
-$as_echo "$func_present" >&6; }
-
-      if test "x$func_present" = "xno"; then
-        as_fn_error $? "$func function required for IPv6 support" "$LINENO" 5
-      fi
-    done
   fi
 fi
 
diff -ruN libetpan-1.7-orig/src/windows/win_init.cpp libetpan-1.7/src/windows/win_init.cpp
--- libetpan-1.7-orig/src/windows/win_init.cpp	2016-05-26 07:27:47.000000000 +0200
+++ libetpan-1.7/src/windows/win_init.cpp	2016-11-24 21:48:50.145880824 +0100
@@ -85,7 +85,7 @@
     WSADATA winsockData;
 
 	void wsocket_init() {
-	    int success = WSAStartup((WORD)0x0101, &winsockData);
+	    int success = WSAStartup(MAKEWORD(2, 2), &winsockData);
 		if (success != 0)
 		{
 			throw "Cannot startup windows sockets api.";
