#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN libetpan-1.7-orig/src/data-types/mailstream_socket.c libetpan-1.7/src/data-types/mailstream_socket.c
--- libetpan-1.7-orig/src/data-types/mailstream_socket.c	2017-04-08 22:46:28.833177620 +0200
+++ libetpan-1.7/src/data-types/mailstream_socket.c	2017-04-08 22:47:58.691407453 +0200
@@ -306,6 +306,7 @@
 static ssize_t mailstream_low_socket_write(mailstream_low * s,
 					   const void * buf, size_t count)
 {
+  ssize_t r;
   struct mailstream_socket_data * socket_data;
 
   socket_data = (struct mailstream_socket_data *) s->data;
@@ -398,8 +399,18 @@
     if (!write_enabled)
       return 0;
   }
-  
-  return send(socket_data->fd, buf, count, 0);
+
+  while (1) {
+    if ((r = send(socket_data->fd, buf, count, 0)) == SOCKET_ERROR) {
+      if (WSAGetLastError() != WSAEWOULDBLOCK) {
+        /* Retry the send() until we get something else than
+         * WSAEWOULDBLOCK. */
+        return r;
+      }
+    } else {
+      return r;
+    }
+  }
 }
 
 
