#! /bin/sh
patch -p0 -f $* < $0
exit $?

--- configure	2016-01-17 10:49:08.646074277 +0100
+++ configure.new	2016-01-21 09:48:17.592507110 +0100
@@ -18271,7 +18271,7 @@
 if test "${with_config_dir+set}" = set; then :
   withval=$with_config_dir; ac_cv_with_config_dir="$withval"
 else
-  ac_cv_with_config_dir=".claws-mail"
+  ac_cv_with_config_dir=""
 fi
 
 if test x"$ac_cv_with_config_dir" = x""; then
