#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -ruN gpgme-1.6.0/src/w32-util.c gpgme-1.6.0-mkstemp/src/w32-util.c
--- gpgme-1.6.0/src/w32-util.c	2015-08-24 16:45:43.000000000 +0200
+++ gpgme-1.6.0-mkstemp/src/w32-util.c	2015-12-08 12:25:32.879506820 +0100
@@ -614,7 +614,7 @@
    does not exist at the time of the call to mkstemp.  TMPL is
    overwritten with the result.  */
 static int
-mkstemp (char *tmpl)
+gpgme_internal_mkstemp (char *tmpl)
 {
   int len;
   char *XXXXXX;
@@ -722,7 +722,7 @@
   if (!tmpname)
     return -1;
   strcpy (stpcpy (tmpname, tmp), "\\gpgme-XXXXXX");
-  *fd = mkstemp (tmpname);
+  *fd = gpgme_internal_mkstemp (tmpname);
   if (fd < 0)
     {
       free (tmpname);
