#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN webkitgtk-2.3.5a-orig/Source/WebCore/rendering/RenderBlock.cpp webkitgtk-2.3.5a/Source/WebCore/rendering/RenderBlock.cpp
--- webkitgtk-2.3.5a-orig/Source/WebCore/rendering/RenderBlock.cpp	2014-02-04 17:38:03.000000000 +0100
+++ webkitgtk-2.3.5a/Source/WebCore/rendering/RenderBlock.cpp	2017-02-05 21:03:51.779067443 +0100
@@ -81,12 +81,6 @@
 
 using namespace HTMLNames;
 
-struct SameSizeAsRenderBlock : public RenderBox {
-    uint32_t bitfields;
-};
-
-COMPILE_ASSERT(sizeof(RenderBlock) == sizeof(SameSizeAsRenderBlock), RenderBlock_should_stay_small);
-
 typedef WTF::HashMap<const RenderBox*, OwnPtr<ColumnInfo>> ColumnInfoMap;
 static ColumnInfoMap* gColumnInfoMap = 0;
 
