#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -ruN webkitgtk-2.3.5-orig/Source/WebKit/gtk/webkit/webkitwebframe.cpp webkitgtk-2.3.5/Source/WebKit/gtk/webkit/webkitwebframe.cpp
--- webkitgtk-2.3.5-orig/Source/WebKit/gtk/webkit/webkitwebframe.cpp	2014-01-21 15:06:51.000000000 +0100
+++ webkitgtk-2.3.5/Source/WebKit/gtk/webkit/webkitwebframe.cpp	2017-05-08 12:39:32.547054972 +0200
@@ -896,8 +896,13 @@
 {
     PrintContext* printContext = reinterpret_cast<PrintContext*>(user_data);
 
+		gdouble x_res = gtk_print_context_get_dpi_x(context);
+		gdouble y_res = gtk_print_context_get_dpi_y(context);
     float width = gtk_print_context_get_width(context);
     float height = gtk_print_context_get_height(context);
+		// force 72 dpi scale for all platforms
+		width /= x_res/72.0;
+		height /= y_res/72.0;
     FloatRect printRect = FloatRect(0, 0, width, height);
 
     printContext->begin(width);
